<!DOCTYPE html>
<html lang="en">

<head>
<link rel="stylesheet" type="text/css" href="../style_related/page_format.css">
<title>10x Chromium Single Cell Multiome ATAC + Gene Expression</title>
</head>
<body>

<h1><a href="https://support.10xgenomics.com/single-cell-multiome-atac-gex" target="_blank"><span style="color:red;">10x Chromium Single Cell Multiome ATAC + Gene Expression</span></a></h1>
<p><info>This kit from 10x Genomics allows simultaneous profiling chromatin accessibility (ATAC) and gene expression (3' mRNA-seq) from the same single cells. You can think of this kit as a combination of scATAC-seq and snRNA-seq. Briefly, the nuclei were used as reaction chambers for Tn5 transposition to tag open chromatin. Then nuclei were isolated using the droplet platform, and mRNA inside the nuclei and tagged open chromatin DNA is captured by oligos on the gel beads. The workflow here is based on the <a href="../data/10X-Genomics/CG000338_ChromiumNextGEM_Multiome_ATAC_GEX_User_Guide_RevF.pdf" target="_blank"><b>revF version of the user guide</b></a> at this time of writing (07-Mar-2024).</info></p>

<p><info><b><span style="color:red;">IMPORTANT:</span></b> Thank <a href="https://github.com/dannyconrad" target="_blank">@dannyconrad</a> for pointing out the new version of the 10x scATAC-seq kit. At this time of writing (2-July-2022), the <a href="../data/10X-Genomics/CG000496_Chromium_NextGEM_SingleCell_ATAC_ReagentKits_v2_UserGuide_RevA.pdf" target="_blank">10x scATAC-seq v2 kit</a> and the <a href="../data/10X-Genomics/CG000338_ChromiumNextGEM_Multiome_ATAC_GEX_User_Guide_RevB.pdf" target="_blank">10x Multiome kit</a> uses the same Tn5 enzyme (<b>PN-2000265/PN-2000272</b>) for the open chromatin tagging. Therefore, it is reasonable to assume that the Tn5 used in the Multiome kit is basically the same as the regular Tn5 that all of us are using in a typical ATAC-seq experiment. To get the tagmented genomic fragments to the beads, a splint or bridging oligo (whatever you want to call it) might be used in the reaction (see details below). In the <a href="https://teichlab.github.io/scg_lib_structs/methods_html/10xChromium_multiome_archive.html" target="_blank">previous Multiome page</a>, which has been archived, I made an assumption that there is a modification of the oligos loaded on the Tn5 enzyme, which might not be the case. The mistake is due to an mis-representation of the Tn5 in the Appendix (Page 79) from the <a href="../data/10X-Genomics/CG000338_ChromiumNextGEM_Multiome_ATAC_GEX_User_Guide_RevB.pdf" target="_blank">Multiome RevB user guide</a>. You can check <a href="https://github.com/Teichlab/scg_lib_structs/issues/15" target="_blank">this thread</a> for details.</info></p>

<p><info>Make sure you use the latest user guide which is <b><a href="../data/10X-Genomics/CG000338_ChromiumNextGEM_Multiome_ATAC_GEX_User_Guide_RevF.pdf" target="_blank">revF</a></b> at this moment. Anyway, the final library structure is definitely correct, just be careful about the the intermediate steps which are based on educational guesses.</info></p>


<br>

<h2>Adapter and primer sequences:</h2>
<seq>
<p>Single Cell Multiome Gel Beads A (PN-2000261). Each individual bead has two types of oligos:</p>
<pre>
<seq>
         For ATAC: |--5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5><cbc>[16-bp cell barcode]</cbc><pe3>CGCGTCTG</pe3> -3'

          For RNA: |--5'- <s5>CTACACGACGCTCTTCCGATCT</s5><cbc>[16-bp cell barcode]</cbc><umi>[12-bp UMI]</umi>TTTTTTTTTTTTTTTTTTTTTTTTTTTTTTVN -3'
</seq>
</pre>
<p>Template Switching Oligo (TSO) (PN-3000228): 5'- <tso>AAGCAGTGGTATCAACGCAGAGTACAT</tso>rGrGrG -3'</p>
<p>Splint/Bridging oligo (presumbably in the Barcoding Reagent Mix PN-2000267): 5'-/phos/- <w1>GACGCTGCCGACGA</w1><pe3>CAGACGCG</pe3> -3'</p>
<p>Pre-Amp Primers (PN-2000271). These should be a mixture of four different primers (fact check ????????) for pre-amplification:</p>
<pre>
<seq>
          ATAC-fwd: 5'- <p5>AATGATACGGCGACCACCGAGA</p5> -3'
          ATAC-rev: 5'- <t7>GTCTCGTGGGCTCGG</t7> -3'

           RNA-fwd: 5'- <s5>CTACACGACGCTCTTCCGATCT</s5> -3'
           RNA-rev: 5'- <tso>AAGCAGTGGTATCAACGCAGAG</tso> -3'
</seq>v
</pre>
<p>cDNA primers (PN-2000089) which are a mixture of two oligos for cDNA amplification:</p>
<pre>
<seq>
           fwd: 5'- <s5>CTACACGACGCTCTTCCGATCT</s5> -3'
           rev: 5'- <tso>AAGCAGTGGTATCAACGCAGAG</tso> -3'
</seq>
</pre>
<p>SI-PCR Primer B (PN-2000128): 5'- <p5>AATGATACGGCGACCACCGAGA</p5> -3'</p>
<p>Sample Index Plate N, Set A (PN-3000427): 5'- <p7>CAAGCAGAAGACGGCATACGAGAT</p7>[8-bp sample index]<t7>GTCTCGTGGGCTCGG</t7> -3'</p>
<p>Illumina P5 adapter: 5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5> -3'</p>
<p>Illumina P7 adapter: 5'- <p7>CAAGCAGAAGACGGCATACGAGAT</p7> -3'</p>
</seq>

<br>

<h2>Step-by-step library generation</h2>
<h3>(1) Nuclei preparation and tagmentation using ATAC Enzyme B (PN-2000265/PN-2000272):</h3>
<img src="../data/10X-Genomics/10x_scATAC_Multiome_Tn5.svg" alt="Tn5 dimer" style="width:800px;height:450px;">

<pre>
<seq>
1.1) mRNA inside nuclei (unaffected):

    5'- XXXXXXXXXXXXX...XXXXXXXXAAAAA...AAAAA -3'

1.2) Open chromatin DNA (three products):

    <i>Product 1 (different ends, the only amplifiable fragments):</i>

        5'- <w1>TCGTCGGCAGCGTC</w1><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me> -3'
                      3'- <me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><t7>GGCTCGGGTGCTCTG</t7> -5'
    
    <ii>Product 2 (same ends, cannot be captured, will be omitted in the next step):</i>

        5'- <t7>GTCTCGTGGGCTCGG</t7><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me> -3'
                       3'- <me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><t7>GGCTCGGGTGCTCTG</t7> -5'
    
    <iii>Product 3 (same ends, can be captured but cannot be amplified, will be omitted):</i>

        5'- <w1>TCGTCGGCAGCGTC</w1><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me> -3'
                      3'- <me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><w1>CTGCGACGGCTGCT</w1> -5'

</seq>
</pre>

<h3>(2) GEM generation and barcoding inside droplet (this is the "37 degree 45mins + 25 degree 30 mins" step):</h3>
<pre>
<seq>
2.1) mRNA:

  2.1.1) Oligo-dT capture mRNA poly-A

     |--5'- <s5>CTACACGACGCTCTTCCGATCT</s5><cbc>[16-bp cell barcode]</cbc><umi>[12-bp UMI]</umi>(T)<sub>30</sub>VN--------->
                                                             3'- (A)<sub>30</sub>BXXXXXXXXXX...XXXXXXXXXXXX -5'
  2.1.2) MMLV adds extra Cs:

     |--5'- <s5>CTACACGACGCTCTTCCGATCT</s5><cbc>[16-bp cell barcode]</cbc><umi>[12-bp UMI]</umi>(T)<sub>30</sub>VXXXXXXXXXXX...XXXXXXXXXXXXCCC
                                                             3'- (A)<sub>30</sub>BXXXXXXXXXX...XXXXXXXXXXXX -5'

  2.1.3) TSO incorporation:

     |--5'- <s5>CTACACGACGCTCTTCCGATCT</s5><cbc>[16-bp cell barcode]</cbc><umi>[12-bp UMI]</umi>(T)<sub>30</sub>VXXXXXXXXXXX...XXXXXXXXXXXCCC------>
                                                             3'- (A)<sub>30</sub>BXXXXXXXXXX...XXXXXXXXXXXXGGG<tso>TACATGAGACGCAACTATGGTGACGAA</tso> -5'
  2.1.4) This will be the fist strand cDNA:

     |--5'- <s5>CTACACGACGCTCTTCCGATCT</s5><cbc>[16-bp cell barcode]</cbc><umi>[12-bp UMI]</umi>(dT)VXXXXXXXXXXX...XXXXXXXXXXXCCC<tso>ATGTACTCTGCGTTGATACCACTGCTT</tso> -3'


2.2) ATAC (simply bead capture with the splint/bridige oligo):

     |--5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5><cbc>[16-bp cell barcode]</cbc><pe3>CGCGTCTG</pe3><w1>TCGTCGGCAGCGTC</w1><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me> -3'
                                                         3'- <pe3>GCGCAGAC</pe3><w1>AGCAGCCGTCGCAG</w1><me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><t7>GGCTCGGGTGCTCTG</t7> -5'
</seq>
</pre>

<h3>(3) Post GEM Incubation Cleanup and Pre-Amplification for 7 cycles using Pre-Amp Primers (PN-2000271):</h3>
<pre>
<seq>
3.1) cDNA:

   5'- <s5>CTACACGACGCTCTTCCGATCT</s5>---------->
|--5'- <s5>CTACACGACGCTCTTCCGATCT</s5><cbc>[16-bp cell barcode]</cbc><umi>[12-bp UMI]</umi>(dT)VXXXXXXXXXXX...XXXXXXXXXXXCCC<tso>ATGTACTCTGCGTTGATACCACTGCTT</tso> -3'
                                                                                     <------------<tso>GAGACGCAACTATGGTGACGAA</tso> -5'

3.2) ATAC:

  3.2.1) The first step of the Pre-Amplification is the 72 degree 5 mins to fill in the gap:

    |--5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5><cbc>[16-bp cell barcode]</cbc><pe3>CGCGTCTG</pe3><w1>TCGTCGGCAGCGTC</w1><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX------>  <me>CTGTCTCTTATACACATCT</me> -3'
                                                       <----<pe3>GCGCAGAC</pe3><w1>AGCAGCCGTCGCAG</w1><me>TCTACACATATTCTCTGTC</me>  <------XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><t7>GGCTCGGGTGCTCTG</t7> -5'

  3.2.2) 7 cycles of amplification:

        5'- <p5>AATGATACGGCGACCACCGAGA</p5>------------->
     |--5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5><cbc>[16-bp cell barcode]</cbc><pe3>CGCGTCTG</pe3><w1>TCGTCGGCAGCGTC</w1><me>AGATGTGTATAAGAGACAG</me>XXX...XXX<me>CTGTCTCTTATACACATCT</me><t7>CCGAGCCCACGAGAC</t7> -3'
        3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5><cbc>[16-bp cell barcode]</cbc><pe3>GCGCAGAC</pe3><w1>AGCAGCCGTCGCAG</w1><me>TCTACACATATTCTCTGTC</me>XXX...XXX<me>GACAGAGAATATGTGTAGA</me><t7>GGCTCGGGTGCTCTG</t7> -5'
                                                                                                                     <------------<t7>GGCTCGGGTGCTCTG</t7> -5'
</seq>
</pre>

<h3>(4) Purification of the Pre-Amp products:</h3>
<pre>
<seq>
4.1) double stranded cDNA:

5'- <s5>CTACACGACGCTCTTCCGATCT</s5><cbc>[16-bp cell barcode]</cbc><umi>[12-bp UMI]</umi>(dT)VXXX...XXXCCC<tso>ATGTACTCTGCGTTGATACCACTGCTT</tso> -3'
3'- <s5>GATGTGCTGCGAGAAGGCTAGA</s5><cbc>[16-bp cell barcode]</cbc><umi>[12-bp UMI]</umi>(pA)BXXX...XXXGGG<tso>TACATGAGACGCAACTATGGTGACGAA</tso> -5'

4.2) ATAC:

5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5><cbc>[16-bp cell barcode]</cbc><pe3>CGCGTCTG</pe3><w1>TCGTCGGCAGCGTC</w1><me>AGATGTGTATAAGAGACAG</me>XXX...XXX<me>CTGTCTCTTATACACATCT</me><t7>CCGAGCCCACGAGAC</t7> -3'
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5><cbc>[16-bp cell barcode]</cbc><pe3>GCGCAGAC</pe3><w1>AGCAGCCGTCGCAG</w1><me>TCTACACATATTCTCTGTC</me>XXX...XXX<me>GACAGAGAATATGTGTAGA</me><t7>GGCTCGGGTGCTCTG</t7> -5'
</seq>
</pre>

<h3>(5) Split into two portions. One portion (35 ul) for Gene Expression and one (40 ul) for ATAC:</h3>
<pre>
<seq>
5.1) cDNA (Gene Expression):

From this step and forward, it is exactly the same as the regular 3' gene expression kit.
Follow Steps (4) to (7) of <b><a href="https://teichlab.github.io/scg_lib_structs/methods_html/10xChromium3.html" target="_blank">this page</a></b> to see how the gene expression library is generated.

5.2) ATAC: amplify using SI-PCR Primer B (PN-2000128) and Sample Index Plate N, Set A (PN-3000427):

5'- <p5>AATGATACGGCGACCACCGAGA</p5>------------->
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5><cbc>[16-bp cell barcode]</cbc><pe3>CGCGTCTG</pe3><w1>TCGTCGGCAGCGTC</w1><me>AGATGTGTATAAGAGACAG</me>XXX...XXX<me>CTGTCTCTTATACACATCT</me><t7>CCGAGCCCACGAGAC</t7> -3'
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5><cbc>[16-bp cell barcode]</cbc><pe3>GCGCAGAC</pe3><w1>AGCAGCCGTCGCAG</w1><me>TCTACACATATTCTCTGTC</me>XXX...XXX<me>GACAGAGAATATGTGTAGA</me><t7>GGCTCGGGTGCTCTG</t7> -5'
                                                                                                          <---------------<t7>GGCTCGGGTGCTCTG</t7>[8-bp sample index]<p7>TAGAGCATACGGCAGAAGACGAAC</p7>-5'
</seq>
</pre>

<h3>(6) Final library structures:</h3>
<pre>
<seq>
6.1) Gene Expression:

5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5><s5>TCTTTCCCTACACGACGCTCTTCCGATCT</s5><cbc>NNNNNNNNNNNNNNNN</cbc><umi>NNNNNNNNNNNN</umi>(dT)VXXX...XXXA<s7>GATCGGAAGAGCACACGTCTGAACTCCAGTCAC</s7>NNNNNNNN<p7>ATCTCGTATGCCGTCTTCTGCTTG</p7> -3'
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5><s5>AGAAAGGGATGTGCTGCGAGAAGGCTAGA</s5><cbc>NNNNNNNNNNNNNNNN</cbc><umi>NNNNNNNNNNNN</umi>(pA)BXXX...XXX<s7>TCTAGCCTTCTCGTGTGCAGACTTGAGGTCAGTG</s7>NNNNNNNN<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
          <p5>Illumina P5</p5>                   <s5>Truseq Read 1</s5>               <cbc>16 bp</cbc>         <umi>12 bp</umi>          cDNA          <s7>Truseq Read 2</s7>                8 bp        <p7>Illumina P7</p7>
                                                                <cbc>cell barcode</cbc>       <umi>UMI</umi>                                                  Sample Index

6.2) ATAC:

5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5><cbc>NNNNNNNNNNNNNNNN</cbc><pe3>CGCGTCTG</pe3><w1>TCGTCGGCAGCGTC</w1><me>AGATGTGTATAAGAGACAG</me>XXX...XXX<me>CTGTCTCTTATACACATCT</me><t7>CCGAGCCCACGAGAC</t7>NNNNNNNN<p7>ATCTCGTATGCCGTCTTCTGCTTG</p7> -3'
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5><cbc>NNNNNNNNNNNNNNNN</cbc><pe3>GCGCAGAC</pe3><w1>AGCAGCCGTCGCAG</w1><me>TCTACACATATTCTCTGTC</me>XXX...XXX<me>GACAGAGAATATGTGTAGA</me><t7>GGCTCGGGTGCTCTG</t7>NNNNNNNN<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
           <p5>Illumina P5</p5>              <cbc>16-bp cell</cbc>    <pe3>spacer</pe3>      <w1>s5</w1>              <me>ME</me>            gDNA           <me>ME</me>               <t7>s7</t7>          8-bp         <p7>Illumina P7</p7>
                                      <cbc>barcode</cbc>                                                                                       sample index
</seq>
</pre>

<br>

<h2>Library sequencing is done separately:</h2>

<h3> (1) For Gene Expression: check the <a href="https://teichlab.github.io/scg_lib_structs/methods_html/10xChromium3.html" target="_blank">10x Chromium Single Cell 3' Solution V2 and V3</a> page.</h3>

<h3> (2) For ATAC: check the <a href="https://teichlab.github.io/scg_lib_structs/methods_html/10xChromium_scATAC.html" target="_blank">10x Chromium Single Cell ATAC</a> page. Note if you use the top strand as the template to sequence the Index 2, which is most likely to happen, make sure the first 8 cycles are dark cycles (the spacer).</h3>

</body>
</html>
