<!DOCTYPE html>
<html lang="en">

<head>
<link rel="stylesheet" type="text/css" href="../style_related/page_format.css">
<title>10x Chromium 5' Gene Expression</title>
</head>
<body>

<h1><a href="https://support.10xgenomics.com/single-cell-vdj" target="_blank"><span style="color:red;">10x Chromium 5' Gene Expression</span></a></h1>
<p><info>The 5' Gene Expression kit is similar to the 3' scRNA-seq methods in general. Instead of using barcoded RT primers on the beads, the 5' Gene Expression kit use a universal RT primer for the reverse transcription, but with barcoded Template Switching Oligos (TSO) on their gel beads. The 5' Gene Expression kit is often coupled with profiling V(D)J from T/B cells (with TCR/BCR primers), but you can simply use this kit for gene expression purposes. Conceptually, this kit is very similar to <a href="https://teichlab.github.io/scg_lib_structs/STRT-seq_family.html" target="_blank"><span style="color:blue;">STRT-seq</span></a>.</info></p>

<p><info>Like <a href="./10xChromium3.html" target="_blank">the 3' Gene Expression kit</a>, there are multiple versions of the 10x Chromium 5' Gene Expression kit based on slightly different chemistry. The library structure of v1 and v2 chemistries are exactly the same, and they are shown on this page. Oligo sequence information is taken from <a href="../data/10X-Genomics/CG000109_AssayConfiguration_VDJ_RevD.pdf" target="_blank"><span style="color:blue;">The 10x Genomics Technical Note</span></a>. The 16-bp cell barcodes are the same as the 3' Gene Expression kit v2, and you can download from here: <a href="../data/10X-Genomics/737K-august-2016.txt.gz" target="_blank">737K-august-2016.txt.gz</a>. This file is copied from <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger" target="blank">Cell Ranger</a> (using Cell Ranger v2.1.0 as an example) /path/to/cellranger-2.1.0/cellranger-cs/2.1.0/tenkit/lib/python/tenkit/barcodes.</info></p>

<p><info><a href="../data/10X-Genomics/CG000733_ChromiumGEM-X_SingleCell5_ReagentKitsv3_UserGuide_RevA.pdf" target="_blank"><b>Around 2023-2024, the v3 chemistry was introduced.</b></a> It has better cell recovery and sensitivity (number of detected genes per cell and TRA/TRB UMI counts) compared to v1 and v2. See their <a href="../data/10X-Genomics/10x_LIT000221_Product_Sheet_GEM-X-Single-Cell-Immune-Profiling_Letter_Digital.pdf" target="_blank">product sheet</a> for a more detailed overview. The cell barcodes have changed in the v3 chemistry, and the file with all 16-bp cell barcodes can be found here: <a href="../data/10X-Genomics/3M-5pgex-jan-2023.txt.gz" target="_blank">3M-5pgex-jan-2023.txt.gz</a>. This file is copied from <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger" target="blank">Cell Ranger</a> (using Cell Ranger v8.0.1 as an example) /path/to/cellranger-8.0.1/lib/python/cellranger/barcodes/. The library structure of 5' v3 is exactly the same as 5' v1 and v2, except that the UMI is 10-bp long in v1 and v2 but 12-bp in v3.</info></p>

<br>

<h2>Adapter and primer sequences:</h2>
<seq>
<p>Beads-TSO:</p>
<pre>
<align class="small">
          V1 (PN-220112) & V2 (PN-1000264): |--5'- <s5>CTACACGACGCTCTTCCGATCT</s5><cbc>[16-bp cell barcode]</cbc><umi>[10-bp UMI]</umi><me>TTTCTTATATrGrGrG</me> -3'

                           V3 (PN-2001129): |--5'- <s5>CTACACGACGCTCTTCCGATCT</s5><cbc>[16-bp cell barcode]</cbc><umi>[12-bp UMI]</umi><me>TTTCTTATATrGrGrG</me> -3'
</align>
</pre>
        
<p>Poly-dT RT primer (PN-2000007): 5'- <tso>AAGCAGTGGTATCAACGCAGAGTAC</tso>TTTTTTTTTTTTTTTTTTTTTTTTTTTTTTVN -3'</p>
<p>cDNA Primer Mix (for cDNA amplification, PN-220106):</p>
<pre>
<align class="small">
        Forward primer: 5'- <s5>CTACACGACGCTCTTCCGATCT</s5> -3'

        Reverse primer: 5'- <tso>AAGCAGTGGTATCAACGCAG</tso> -3'
</align>
</pre>
<p>Illumina Truseq Read 1 primer: 5'- <p5>ACAC</p5><s5>TCTTTCCCTACACGACGCTCTTCCGATCT</s5> -3'</p>
<p>Illumina Truseq Read 2 primer: 5'- <s7>GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT</s7> -3'</p>
<p>Truseq adapter (double stranded DNA with a T overhang, PN-220026):</p>
<pre>
<align class="small">
        5'-  <s7>GATCGGAAGAGCACACGTCTGAACTCCAGTCAC</s7> -3'
        3'- <s7>TCTAGCCTTCTCG</s7> -5'
</align>
</pre>
<p>Library PCR primer 1 (PN-220111): 5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5><s5>TCTTTCCCTACACGACGCTC</s5> -3'</p>
<p>Library PCR primer 2 (PN-220103): 5'- <p7>CAAGCAGAAGACGGCATACGAGAT</p7>[8-bp sample index]<s7>GTGACTGGAGTTCAGACGTGT</s7> -3'</p>
<p>Sample index sequencing primer: 5'- <s7>GATCGGAAGAGCACACGTCTGAACTCCAGTCAC</s7> -3'</p>
<p>Illumina P5 adapter: 5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5> -3'</p>
<p>Illumina P7 adapter: 5'- <p7>CAAGCAGAAGACGGCATACGAGAT</p7> -3'</p>
</seq>

<br>

<h2>Step-by-step library generation</h2>
<h3>(1) Reverse transcription with Poly-dT RT primer using MMLV:</h3>
<pre>
<seq>
                   <--------NV(T)<sub>30</sub><tso>CATGAGACGCAACTATGGTGACGAA</tso> -5'
    5'- XXXXXXXXXXXXXXXXXXXXXB(A)<sub>30</sub>
</seq>
</pre>

<h3>(2) The terminal transferase activity of MMLV adds extra Cs:</h3>
<pre>
<seq>
     CCCXXXXXXXXXXXXXXXXXXXXXNV(T)<sub>30</sub><tso>CATGAGACGCAACTATGGTGACGAA</tso> -5'
    5'- XXXXXXXXXXXXXXXXXXXXXXB(A)<sub>30</sub>
</seq>
</pre>

<h3>(3) cDNA capture by gel bead barcoded TSO:</h3>
<pre>
<seq>
                                                          <-----------CCCXXXXXXXXXXXXXXXXXXXXXNV(T)<sub>30</sub><tso>CATGAGACGCAACTATGGTGACGAA</tso> -5'
|--5'- <s5>CTACACGACGCTCTTCCGATCT</s5><cbc>[16-bp cell barcode]</cbc><umi>[10-bp UMI]</umi><me>TTTCTTATATGGG</me>XXXXXXXXXXXXXXXXXXXXXXB(A)<sub>30</sub>------->
</seq>
</pre>

<h3>(4) Adding cDNA Primer Mix to amplify full length cDNA:</h3>
<pre>
<seq>
   5'- <s5>CTACACGACGCTCTTCCGATCT</s5>-------->
|--5'- <s5>CTACACGACGCTCTTCCGATCT</s5><cbc>[16-bp cell barcode]</cbc><umi>[10-bp UMI]</umi><me>TTTCTTATATGGG</me>XXXXXXXXX...XXXXXXXXXB(pA)<tso>GTACTCTGCGTTGATACCACTGCTT</tso> -3'
   3'- <s5>GATGTGCTGCGAGAAGGCTAGA</s5><cbc>[16-bp cell barcode]</cbc><umi>[10-bp UMI]</umi><me>AAAGAATATACCC</me>XXXXXXXXX...XXXXXXXXXV(dT)<tso>CATGAGACGCAACTATGGTGACGAA</tso> -5'
                                                                                               <--------<tso>GACGCAACTATGGTGACGAA</tso> -5'
</seq>
</pre>

<h3>(5) Use Fragmentase to fragment cDNA and perform A-tailing:</h3>
<pre>
<seq>
<i>Product 1 (TSO, cell barcode, UMI plus 5'-end of cDNA):</i>


5'-   <s5>CTACACGACGCTCTTCCGATCT</s5><cbc>[16-bp cell barcode]</cbc><umi>[10-bp UMI]</umi><me>TTTCTTATATGGG</me>XXXXXXXXX...XXXXXXXXX*A -3'
3'- A*<s5>GATGTGCTGCGAGAAGGCTAGA</s5><cbc>[16-bp cell barcode]</cbc><umi>[10-bp UMI]</umi><me>AAAGAATATACCC</me>XXXXXXXXX...XXXXXXXXX   -5'


<i>Product 2 (middle of cDNA):</i>


5'-   XXXXXXXXXXXXXXXXXXXXXXXXX...XXXXXXXXXXXXXXXXXXXXXXXXX*A -3'
3'- A*XXXXXXXXXXXXXXXXXXXXXXXXX...XXXXXXXXXXXXXXXXXXXXXXXXX   -5'


<i>Product 3 (3' of cDNA plus Poly-dT RT sequence):</i>


5'-   <tso>AAGCAGTGGTATCAACGCAGAGTAC</tso>(dT)VXXXXXXXXX...XXXXXXXXX*A -3'
3'- A*<tso>TTCGTCACCATAGTTGCGTCTCATG</tso>(pA)BXXXXXXXXX...XXXXXXXXX   -5'

</seq>
</pre>

<h3>(6) Add double stranded Illumina Truseq adapter with a T overhang (PN-220026) for ligation:</h3>
<pre>
<seq>
<i>Product 1 (I assume the 5' end of Poly-dT RT primer is blocked, so the adapter can only be ligated to the cDNA end. This is the only ampliable fragment):</i>


5'-   <s5>CTACACGACGCTCTTCCGATCT</s5><cbc>[16-bp cell barcode]</cbc><umi>[10-bp UMI]</umi><me>TTTCTTATATGGG</me>XXX...XXXA<s7>GATCGGAAGAGCACACGTCTGAACTCCAGTCAC</s7> -3'
3'- A*<s5>GATGTGCTGCGAGAAGGCTAGA</s5><cbc>[16-bp cell barcode]</cbc><umi>[10-bp UMI]</umi><me>AAAGAATATACCC</me>XXX...XXX<s7>TCTAGCCTTCTCG</s7> -5'


<i>Product 2 (will not amplify efficiently due to semi-suppressive PCR??? not really sure about this):</i>


5'- <s7>                     GCTCTTCCGATCT</s7>XXXXXXXXXXXXXXXXXXXX...XXXXXXXXXXXXXXXXXXXXA<s7>GATCGGAAGAGCACACGTCTGAACTCCAGTCAC</s7> -3'
3'- <s7>CACTGACCTCAAGTCTGCACACGAGAAGGCTAG</s7>AXXXXXXXXXXXXXXXXXXXX...XXXXXXXXXXXXXXXXXXXX<s7>TCTAGCCTTCTCG</s7> -5'


<i>Product 3 (I assume the 5' end of TSO is blocked. This produce is not amplifiable due to the use of the specific primers for amplification, see the next step):</i>


5'-   <tso>AAGCAGTGGTATCAACGCAGAGTAC</tso>(dT)VXXXXXXXXX...XXXXXXXXXA<s7>GATCGGAAGAGCACACGTCTGAACTCCAGTCAC</s7> -3'
3'- A*<tso>TTCGTCACCATAGTTGCGTCTCATG</tso>(pA)BXXXXXXXXX...XXXXXXXXX<s7>TCTAGCCTTCTCG</s7> -5'


</seq>
</pre>

<h3>(7) Add Library PCR Primers 1 & 2 to amplify library:</h3>
<pre>
<align class="long">
5'-  <p5>AATGATACGGCGACCACCGAGATCTACAC</p5><s5>TCTTTCCCTACACGACGCTC</s5>--------->
                                   5'-   <s5>CTACACGACGCTCTTCCGATCT</s5><cbc>[16-bp cell barcode]</cbc><umi>[10-bp UMI]</umi><me>TTTCTTATATGGG</me>XXX...XXXA<s7>GATCGGAAGAGCACACGTCTGAACTCCAGTCAC</s7> -3'
                                   3'- A*<s5>GATGTGCTGCGAGAAGGCTAGA</s5><cbc>[16-bp cell barcode]</cbc><umi>[10-bp UMI]</umi><me>AAAGAATATACCC</me>XXX...XXX<s7>TCTAGCCTTCTCG</s7>                      -5'
                                                                                                                     <-----------<s7>TGTGCAGACTTGAGGTCAGTG</s7>[8-bp sample index]<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(8) Final library structure:</h3>
<h4>V1 and V2 library:</h4>
<pre>
<align class="long">
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5><s5>TCTTTCCCTACACGACGCTCTTCCGATCT</s5><cbc>NNNNNNNNNNNNNNNN</cbc><umi>NNNNNNNNNN</umi><me>TTTCTTATATGGG</me>XXX...XXXA<s7>GATCGGAAGAGCACACGTCTGAACTCCAGTCAC</s7>NNNNNNNN<p7>ATCTCGTATGCCGTCTTCTGCTTG</p7> -3'
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5><s5>AGAAAGGGATGTGCTGCGAGAAGGCTAGA</s5><cbc>NNNNNNNNNNNNNNNN</cbc><umi>NNNNNNNNNN</umi><me>AAAGAATATACCC</me>XXX...XXX<s7>TCTAGCCTTCTCGTGTGCAGACTTGAGGTCAGTG</s7>NNNNNNNN<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
          <p5>Illumina P5</p5>                   <s5>Truseq Read 1</s5>               <cbc>16 bp</cbc>       <umi>10 bp</umi>                   cDNA          <s7>Truseq Read 2</s7>                8 bp        <p7>Illumina P7</p7>
                                                                <cbc>cell barcode</cbc>     <umi>UMI</umi>                                                           Sample Index
</align>
</pre>

<h4>V3 library:</h4>
<pre>
<align class="long">
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5><s5>TCTTTCCCTACACGACGCTCTTCCGATCT</s5><cbc>NNNNNNNNNNNNNNNN</cbc><umi>NNNNNNNNNNNN</umi><me>TTTCTTATATGGG</me>XXX...XXXA<s7>GATCGGAAGAGCACACGTCTGAACTCCAGTCAC</s7>NNNNNNNN<p7>ATCTCGTATGCCGTCTTCTGCTTG</p7> -3'
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5><s5>AGAAAGGGATGTGCTGCGAGAAGGCTAGA</s5><cbc>NNNNNNNNNNNNNNNN</cbc><umi>NNNNNNNNNNNN</umi><me>AAAGAATATACCC</me>XXX...XXX<s7>TCTAGCCTTCTCGTGTGCAGACTTGAGGTCAGTG</s7>NNNNNNNN<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
          <p5>Illumina P5</p5>                   <s5>Truseq Read 1</s5>               <cbc>16 bp</cbc>        <umi>12 bp</umi>                   cDNA          <s7>Truseq Read 2</s7>                8 bp        <p7>Illumina P7</p7>
                                                                <cbc>cell barcode</cbc>      <umi>UMI</umi>                                                           Sample Index
</align>
</pre>

<br>

<h2>Library sequencing:</h2>

<h3>(1) Add Truseq Read 1 primer to sequence the first read (bottom strand as template, sequence 16-bp cell barcode and UMI, 26 cycles for V1 and V2, 28 cycles for V3):</h3>
<pre>
<align class="long">
                         5'- <p5>ACAC</p5><s5>TCTTTCCCTACACGACGCTCTTCCGATCT</s5>------------------------>
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5><s5>AGAAAGGGATGTGCTGCGAGAAGGCTAGA</s5><cbc>NNNNNNNNNNNNNNNN</cbc><umi>NNNNNNNNNN</umi><me>AAAGAATATACCC</me>XXX...XXX<s7>TCTAGCCTTCTCGTGTGCAGACTTGAGGTCAGTG</s7>NNNNNNNN<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(2) Add Sample Index sequencing primer to sequence the sample index (bottom strand as template):</h3>
<pre>
<align class="long">
                                                                                                           5'- <s7>GATCGGAAGAGCACACGTCTGAACTCCAGTCAC</s7>------->
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5><s5>AGAAAGGGATGTGCTGCGAGAAGGCTAGA</s5><cbc>NNNNNNNNNNNNNNNN</cbc><umi>NNNNNNNNNN</umi><me>AAAGAATATACCC</me>XXX...XXX<s7>TCTAGCCTTCTCGTGTGCAGACTTGAGGTCAGTG</s7>NNNNNNNN<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(3) Cluster regeneration, add Truseq Read 2 primer to sequence the second read (top strand as template, sequence cDNA, 98 cycles):</h3>
<pre>
<align class="long">
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5><s5>TCTTTCCCTACACGACGCTCTTCCGATCT</s5><cbc>NNNNNNNNNNNNNNNN</cbc><umi>NNNNNNNNNN</umi><me>TTTCTTATATGGG</me>XXX...XXXA<s7>GATCGGAAGAGCACACGTCTGAACTCCAGTCAC</s7>NNNNNNNN<p7>ATCTCGTATGCCGTCTTCTGCTTG</p7> -3'
                                                                                                        <-----<s7>TCTAGCCTTCTCGTGTGCAGACTTGAGGTCAGTG</s7> -5'
</align>
</pre>

</body>
</html>
