<!DOCTYPE html>
<html lang="en">

<head>
<link rel="stylesheet" type="text/css" href="../style_related/page_format.css">
<title>CH-ATAC-seq</title>
</head>

<body>

<h1><a href="https://doi.org/10.1016/j.devcel.2024.01.015" target="_blank">CH-ATAC-seq</a></h1>

<p><info>CH-ATAC-seq is like <a href="https://teichlab.github.io/scg_lib_structs/methods_html/sci-ATAC-seq3.html" target="_blank">sci-ATAC-seq3</a> that uses combinatorial indexing to achieve single cell chromatin accessibility profiling. They difference is that CH-ATAC-seq uses barcoded Tn5 with to add the first-level barcode. Then subsequent rounds of barcodes are added through hybridisation. The oligo sequences here are taken from the <a href="../data/CH-ATAC-seq/CH-ATAC-seq_SupplementaryTable1.xlsx" target="_blank">Supplementary Table 1</a> from the paper, which is designed to work on the BGI system.</info></p>

<br>

<h2>Adapter and primer sequences:</h2>
<seq>
<p>Tn5 binding site 19-bp Mosaic End (ME) bottom: 5'-/Phos/ <me>CTGTCTCTTATACACATCT</me> -3'</p>
<p>Tn5_barcode_primer_{1..384}: 5'- <r1>AAGCAGTGGTATCAACGCAGAGT</r1><cbc>[10-bp Tn5 barcode]</cbc><me>AGATGTGTATAAGAGACAG</me> -3'</p>
<p>Tn5_Primer_C_oligo: 5'- <s7>GTCTCGTGGGCTCGG</s7><me>AGATGTGTATAAGAGACAG</me> -3'</p>
<p>HY_head_oligo: 5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me> -3'</p>
<p>Barcoded_HY_oligo_{1..768}: 5'- <r1>ACTCTGCGTTGATACCACTGCTT</r1><cbc>[10-bp HY barcode]</cbc><me>CTGTCTCTTATACACATCT</me><s5>GACGCTGCCGACGA</s5> -3'</p>
<p>Block_tail_primer_oligo: 5'- <r1>AAGCAGTGGTATCAACGCAGAGT</r1> -3'</p>
<p>PCR P5 primer: 5'- <p5>GAACGACATGGCTACGATCCGACTT</p5><s5>TCGTCGGCAGCGTC</s5> -3'</p>
<p>MGI_P7_index_{1..96}: 5'- <p7>TGTGAGCCAAGGAGTTGTTGTCTTC</p7>[10-bp i7]<s7>GTCTCGTGGGCTCGG</s7> -3'</p>
<p>MGI_P5: 5'- <p5>GAACGACATGGCTACGATCCGACTT</p5> -3' </p>
<p>MGI_P7: 5'- <p7>TGTGAGCCAAGGAGTTGTTGTCTTC</p7> -3' </p>
<p>Read 1 sequencing primer: 5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me> -3'</p>
<p>Index i7 sequencing primer: 5'- <me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7> -3'</p>
<p>Read 2 sequencing primer: 5'- <s7>GTCTCGTGGGCTCGG</s7><me>AGATGTGTATAAGAGACAG</me> -3'</p>
</seq>


<br>

<h2>Step-by-step library generation</h2>
<h3>(1) Anneal "Tn5_ME+Tn5_barcode_primer" and "Tn5_ME+Tn5_Primer_C_oligo" and use those to assemble indexed Tn5 transposome:</h3>
<img src="../data/CH-ATAC-seq/CH-ATAC-seq_Tn5.svg" alt="Tn5 dimer" style="width:800px">

<h3>(2) Bulk nuclei tagging by the indexed Tn5 shown above. There are 3 different products (will create 9 bp gap):</h3>
<pre>
<seq>
<i>Product 1 (purple at both ends, not amplifiable due to <a href="http://www.nature.com/nmeth/journal/v7/n7/full/nmeth.1470.html" target="_blank">semi-suppressiev PCR</a>, omitted afterwards):</i>

5'- <r1>AAGCAGTGGTATCAACGCAGAGT</r1><cbc>[10-bp Tn5 barcode]</cbc><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me>
                                              <me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><cbc>[10-bp Tn5 barcode]</cbc><r1>TGAGACGCAACTATGGTGACGAA</r1> -5'


<i>Product 2 (orange at both ends, cannot be ligated, omitted afterwards):</i>

5'- <s7>GTCTCGTGGGCTCGG</s7><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me>
                   <me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7> -5'


<i>Product 3 (different ends, the only usable fragment):</i>

5'- <r1>AAGCAGTGGTATCAACGCAGAGT</r1><cbc>[10-bp Tn5 barcode]</cbc><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me>
                                              <me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7> -5'
</seq>
</pre>

<h3>(3) Pool nuclei and redistribute to new plates for hybridisation barcode addition:</h3>
<pre>
<seq>
<i>(3.1) Anneal HY_head_oligo with Barcoded_HY_oligo_{1..768}:</i>

5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>
3'- <s5>AGCAGCCGTCGCAG</s5><me>TCTACACATATTCTCTGTC</me><cbc>[10-bp HY barcode]</cbc><r1>TTCGTCACCATAGTTGCGTCTCA</r1> -5'


<i>(3.2) Hybridise to transposed nuclei and ligate the HY barcode:</i>

5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>                  <r1>AAGCAGTGGTATCAACGCAGAGT</r1><cbc>[10-bp Tn5 barcode]</cbc><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me>
3'- <s5>AGCAGCCGTCGCAG</s5><me>TCTACACATATTCTCTGTC</me><cbc>[10-bp HY barcode]</cbc><r1>TTCGTCACCATAGTTGCGTCTCA</r1> -5'               <me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7> -5'

</seq>
</pre>

<h3>(4) Add Block_tail_primer_oligo to block unused HY oligos, pool nuclei, redistribute to new plates and perform gap fill-in:</h3>
<pre>
<seq>
5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me><cbc>[10-bp HY barcode]</cbc><r1>AAGCAGTGGTATCAACGCAGAGT</r1><cbc>[10-bp Tn5 barcode]</cbc><me>AGATGTGTATAAGAGACAG</me>XXX...XXX<me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7>
    <s5>AGCAGCCGTCGCAG</s5><me>TCTACACATATTCTCTGTC</me><cbc>[10-bp HY barcode]</cbc><r1>TTCGTCACCATAGTTGCGTCTCA</r1><cbc>[10-bp Tn5 barcode]</cbc><me>TCTACACATATTCTCTGTC</me>XXX...XXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7> -5'
</seq>
</pre>

<h3>(5) Lyse cells and use PCR P5 primer + MGI_P7_index_{1..96} primer to amplify the library:</h3>
<pre>
<seq>
5'- <p5>GAACGACATGGCTACGATCCGACTT</p5><s5>TCGTCGGCAGCGTC</s5>---------->
                         5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me><cbc>[10-bp HY barcode]</cbc><r1>AAGCAGTGGTATCAACGCAGAGT</r1><cbc>[10-bp Tn5 barcode]</cbc><me>AGATGTGTATAAGAGACAG</me>XXX...XXX<me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7>
                             <s5>AGCAGCCGTCGCAG</s5><me>TCTACACATATTCTCTGTC</me><cbc>[10-bp HY barcode]</cbc><r1>TTCGTCACCATAGTTGCGTCTCA</r1><cbc>[10-bp Tn5 barcode]</cbc><me>TCTACACATATTCTCTGTC</me>XXX...XXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7> -5'
                                                                                                                                                               <---------<s7>GGCTCGGGTGCTCTG</s7>[10-bp i7]<p7>CTTCTGTTGTTGAGGAACCGAGTGT</p7> -5'
</seq>
</pre>

<h3>(6) Final library structure:</h3>
<pre>
<align class="long">
5'- <p5>GAACGACATGGCTACGATCCGACTT</p5><s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me><cbc>NNNNNNNNNN</cbc><r1>AAGCAGTGGTATCAACGCAGAGT</r1><cbc>NNNNNNNNNN</cbc><me>AGATGTGTATAAGAGACAG</me>XXX...XXX<me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7>NNNNNNNNNN<p7>GAAGACAACAACTCCTTGGCTCACA</p7> -3'
3'- <p5>CTTGCTGTACCGATGCTAGGCTGAA</p5><s5>AGCAGCCGTCGCAG</s5><me>TCTACACATATTCTCTGTC</me><cbc>NNNNNNNNNN</cbc><r1>TTCGTCACCATAGTTGCGTCTCA</r1><cbc>NNNNNNNNNN</cbc><me>TCTACACATATTCTCTGTC</me>XXX...XXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7>NNNNNNNNNN<p7>CTTCTGTTGTTGAGGAACCGAGTGT</p7> -5'
              <p5>MGI P5</p5>               <s5>s5</s5>             <me>ME</me>            <cbc>10-bp</cbc>         <r1>HY linker</r1>          <cbc>10-bp</cbc>           <me>ME</me>           gDNA          <me>ME</me>                <s7>s7</s7>        10-bp i7          <p7>MGI P7</p7>
                                                             <cbc>HY barcode</cbc>                       <cbc>Tn5 barcode</cbc>
</align>
</pre>

<br>

<h2>Library sequencing (not sure how MGI sequencing works, so this section is just based on guesses):</h2>

<h3>(1) Add Read 1 sequencing primer to sequence the first read (bottom strand as template, 100 cycles,  with dark cycles from 11-33 and 44-62):</h3>
<pre>
<align class="long">
                         5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>----------ooooooooooooooooooooooo----------ooooooooooooooooooo--->
3'- <p5>CTTGCTGTACCGATGCTAGGCTGAA</p5><s5>AGCAGCCGTCGCAG</s5><me>TCTACACATATTCTCTGTC</me><cbc>NNNNNNNNNN</cbc><r1>TTCGTCACCATAGTTGCGTCTCA</r1><cbc>NNNNNNNNNN</cbc><me>TCTACACATATTCTCTGTC</me>XXX...XXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7>NNNNNNNNNN<p7>CTTCTGTTGTTGAGGAACCGAGTGT</p7> -5'
</align>
</pre>

<h3>(2) Add Index 1 sequencing primer to sequence the i7 index (bottom strand as template, 10 cycles):</h3>
<pre>
<align class="long">
                                                                                                                                 5'- <me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7>--------->
3'- <p5>CTTGCTGTACCGATGCTAGGCTGAA</p5><s5>AGCAGCCGTCGCAG</s5><me>TCTACACATATTCTCTGTC</me><cbc>NNNNNNNNNN</cbc><r1>TTCGTCACCATAGTTGCGTCTCA</r1><cbc>NNNNNNNNNN</cbc><me>TCTACACATATTCTCTGTC</me>XXX...XXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7>NNNNNNNNNN<p7>CTTCTGTTGTTGAGGAACCGAGTGT</p7> -5'
</align>
</pre>

<h3>(3) Add Read 2 sequencing primer to sequence the second read (top strand as template, 100 cycles):</h3>
<pre>
<align class="long">
5'- <p5>GAACGACATGGCTACGATCCGACTT</p5><s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me><cbc>NNNNNNNNNN</cbc><r1>AAGCAGTGGTATCAACGCAGAGT</r1><cbc>NNNNNNNNNN</cbc><me>AGATGTGTATAAGAGACAG</me>XXX...XXX<me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7>NNNNNNNNNN<p7>GAAGACAACAACTCCTTGGCTCACA</p7> -3'
                                                                                                                             <-------<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7> -5'
</align>
</pre>

<br>

</body>
</html>
