<!DOCTYPE html>
<html lang="en">

<head>
<link rel="stylesheet" type="text/css" href="../style_related/page_format.css">
<title>HyDrop-RNA</title>
</head>
<body>

<h1><a href="https://elifesciences.org/articles/73971" target="_blank"><span style="color:red;">HyDrop-RNA</span></a></h1>
<p><span style="font-size:1.1em;">The HyDrop protocols were published in eLife on Feb 23th, 2022 (<a href="https://elifesciences.org/articles/73971" target="_blank">De Rop <i>et al.</i>  eLife 11:e73971</a>). The authors developed an open source droplet system, HyDrop, as a hybrid method between <a href="https://teichlab.github.io/scg_lib_structs/methods_html/inDrop.html" target="_blank">inDrop</a> and <a href="https://teichlab.github.io/scg_lib_structs/methods_html/Drop-seq.html" target="_blank">Drop-Seq</a>. They optimised various steps during the library construction procedures and utilised a dissolvable hydrogel beads to improve barcoded primer release and diffusion.</span></p>

<p><span style="font-size:1.1em;">Both scRNA-seq and scATAC-seq can be performed on the HyDrop platform. In this page, Hydrop-RNA is presented. This page is basically a recreation of the Supplementary Files <a href="https://cdn.elifesciences.org/articles/73971/elife-73971-supp1-v4.docx" target="_blank">1</a> & <a href="https://cdn.elifesciences.org/articles/73971/elife-73971-supp3-v4.docx" target="_blank">3</a> from the publication. Detailed protocol can be found here: <a href="https://dx.doi.org/10.17504/protocols.io.b4xwqxpe" target="_blank">protocols.io</a>.</span></p>

<p><span style="font-size:1.1em;">For scATAC-seq on the HyDrop platform, go to the <a href="https://teichlab.github.io/scg_lib_structs/methods_html/HyDrop_ATAC.html" target="_blank">HyDrop-ATAC page</a> for a detailed step-by-step illustration.</span></p>

<br></br>

<h2>Adapter and primer sequences:</h2>

<p><b><span style="color:red; font-size:1.1em;">*</span></b> <span style="font-size:1.1em;">The complete full list of oligo sequences can be found in this spreadsheet from their protocols.io page: <a href="../data/20210712_supp_methods_table_hydrop_oligonucleotide_list.xlsx" target="_blank">20210712_supp_methods_table_hydrop_oligonucleotide_list.xlsx</a>.</span></p>

<br></br>

<h3>Sequence used during the experiment:</h3>
<seq>
<p>Barcoded beads-oligo: |--5'- /5Acryd//iThioMC6-D/TTTTTTTTAATACGACTCACTATAGGG<tso>AAGCAGTGGTATCAACGCAGAGT</tso>AC<cbc>[10-bp barcode1]</cbc>CAGCTACTGC<cbc>[10-bp barcode2]</cbc>CGAGTACCCT<cbc>[10-bp barcode3]</cbc><umi>[8-bp UMI]</umi>TTTTTTTTTTTTTTTTTTTTTTTTTTTTTT -3'</p>
<p>Template Switching Oligo (TSO): 5'- <tso>AAGCAGTGGTATCAACGCAGAGT</tso>GAATrGrGrG -3'</p>
<p>TSO-P: 5'- <tso>AAGCAGTGGTATCAACGCAGAGT</tso> -3'</p>
<p>NEB Hairpin adapter:</p>
<pre>
    <align class="small">
            <s5>ACACTCTTTCCCTACACGAC</s5>
            |                   <s5>GCTCTTCCGATCT</s5> -3'
            U                   <s5>CGAGAAGGCTAG</s5> -5'
            \<s5>CTGACCTCAAGTCTGCACA</s5>
    </align>
</pre>
<p>HYi5_TruSeq: 5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>[10-bp index]<s5>ACACTCTTTCCCTACACGACGCT</s5> -3'</p>
<p>HYi7: 5'- <p7>CAAGCAGAAGACGGCATACGAGAT</p7>[10-bp index]CTGTCCGCGG<tso>AAGCAGTGGTATCAACGCAGAGT</tso>AC - 3'</p>
<p>TruSeq Read 1: 5'- <s5>ACACTCTTTCCCTACACGACGCTCTTCCGATCT</s5> -3'</p>
<p>HyDrop_CustSeq_Short: 5'- GT<tso>ACTCTGCGTTGATACCACTGCTT</tso>CCGCGGACAG -3'</p>
<p>HyDrop_CustSeq_R2: 5'- CTGTCCGCGG<tso>AAGCAGTGGTATCAACGCAGAGT</tso>AC -3'</p>
<p>Index 2 Sequencing Primer: 5'- <s5>AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT</s5> -3'</p>
<p>Illumina P5 adapter: 5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5> -3'</p>
<p>Illumina P7 adapter: 5'- <p7>CAAGCAGAAGACGGCATACGAGAT</p7> -3'</p>
</seq>
<h3>Sequence used for barcoded bead generation (before the experiment):</h3>
<seq>
<p>Acrydite_primer: 5'- /5Acryd//iThioMC6-D/TTTTTTTTAATACGACTCACTATAGGG<tso>AAGCAGTGGTATCAACGCAGAGT</tso>AC -3'</p>
<p>20200130_plate-1-96: 5' - GCAGTAGCTG<cbc>[10-bp barcode1]</cbc>GT<tso>ACTCTGCG</tso> - 3'</p> 
<p>20200130_plate-2-96: 5' - AGGGTACTCG<cbc>[10-bp barcode2]</cbc>GCAGTAGCTG - 3'</p>
<p>20200130_plate-3-96-RNAseq: 5'- AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA<umi>[8-bp UMI]</umi><cbc>[10-bp barcode3]</cbc>AGGGTACTCG -3'</p>
<p><b><span style="color:red;">*</span></b> The following oligos are only used for QC of the beads, not really used in the real experiments, so they are not drawn in the workflow in this page:</p>
<p>Anti-Acrydite_FAM: 5'- /56-FAM/TTTTTGT<tso>ACTCTGCGTTGATACCAC</tso> -3'</p>
<p>Anti-RNA_FAM: 5'- /56-FAM/AAAAAAAAAAAAAAAAAAAA -3'</p>
<p>Anti-BC1_FAM: 5'- /56-FAM/TTTTTCTATCCGTCAGTAC -3'</p>
<p>Anti-BC2_FAM: 5'- /56-FAM/TTTTTACACGTTGTGGCAG -3'</p>
<p>Anti-BC3_FAM: 5'- /56-FAM/TTTTTCTCCTATCATAGGG -3'</p>
</seq>

<br></br>

<h2>Step-by-step generation of barcoded beads:</h2>
<h3>(1) Form dissolvable acrylimide gel beads together with Acrydite_primer:</h3>
<pre>
<seq>
|--5'- /5Acryd//iThioMC6-D/TTTTTTTTAATACGACTCACTATAGGG<tso>AAGCAGTGGTATCAACGCAGAGT</tso>AC -3'
</seq>
</pre>

<h3>(2) Split the gel beads into wells in 20200130_plate-1-96, and perform extension:</h3>
<pre>
<seq>
|--5'- /5Acryd//iThioMC6-D/TTTTTTTTAATACGACTCACTATAGGG<tso>AAGCAGTGGTATCAACGCAGAGT</tso>AC---->
                                                                <----<tso>GCGTCTCA</tso>TG<cbc>[10-bp barcode1]</cbc>GTCGATGACG -5'
</seq>
</pre>

<h3>(3) This is the product after the first extension:</h3>
<pre>
<seq>
|--5'- /5Acryd//iThioMC6-D/TTTTTTTTAATACGACTCACTATAGGG<tso>AAGCAGTGGTATCAACGCAGAGT</tso>AC<cbc>[10-bp barcode1]</cbc>CAGCTACTGC -3'
                       3'- AAAAAAAATTATGCTGAGTGATATCCC<tso>TTCGTCACCATAGTTGCGTCTCA</tso>TG<cbc>[10-bp barcode1]</cbc>GTCGATGACG -5'
</seq>
</pre>

<h3>(4) Pooling, denature by NaOH, and get rid of top strand:</h3>
<pre>
<seq>
|--5'- /5Acryd//iThioMC6-D/TTTTTTTTAATACGACTCACTATAGGG<tso>AAGCAGTGGTATCAACGCAGAGT</tso>AC<cbc>[10-bp barcode1]</cbc>CAGCTACTGC -3'
</seq>
</pre>

<h3>(5) Split again into wells in 20200130_plate-2-96, and perform extension:</h3>
<pre>
<align class="long">
|--5'- /5Acryd//iThioMC6-D/TTTTTTTTAATACGACTCACTATAGGG<tso>AAGCAGTGGTATCAACGCAGAGT</tso>AC<cbc>[10-bp barcode1]</cbc>CAGCTACTGC---->
                                                                                          <----GTCGATGACG<cbc>[10-bp barcode2]</cbc>GCTCATGGGA -5'
</align>
</pre>

<h3>(6) This is the product of the second extension:</h3>
<pre>
<align class="long">
|--5'- /5Acryd//iThioMC6-D/TTTTTTTTAATACGACTCACTATAGGG<tso>AAGCAGTGGTATCAACGCAGAGT</tso>AC<cbc>[10-bp barcode1]</cbc>CAGCTACTGC<cbc>[10-bp barcode2]</cbc>CGAGTACCCT -3'
                       3'- AAAAAAAATTATGCTGAGTGATATCCC<tso>TTCGTCACCATAGTTGCGTCTCA</tso>TG<cbc>[10-bp barcode1]</cbc>GTCGATGACG<cbc>[10-bp barcode2]</cbc>GCTCATGGGA -5'
</align>
</pre>

<h3>(7) Pool again, denature by NaOH, get rid of top strand:</h3>
<pre>
<align class="long">
|--5'- /5Acryd//iThioMC6-D/TTTTTTTTAATACGACTCACTATAGGG<tso>AAGCAGTGGTATCAACGCAGAGT</tso>AC<cbc>[10-bp barcode1]</cbc>CAGCTACTGC<cbc>[10-bp barcode2]</cbc>CGAGTACCCT -3'
</align>
</pre>

<h3>(8) Split again into wells in 20200130_plate-3-96-RNAseq, and perform extension:</h3>
<pre>
<align class="long">
|--5'- /5Acryd//iThioMC6-D/TTTTTTTTAATACGACTCACTATAGGG<tso>AAGCAGTGGTATCAACGCAGAGT</tso>AC<cbc>[10-bp barcode1]</cbc>CAGCTACTGC<cbc>[10-bp barcode2]</cbc>CGAGTACCCT---->
                                                                                                                    <----GCTCATGGGA<cbc>[10-bp barcode3]</cbc><umi>[8-bp UMI]</umi>AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA -5'
</align>
</pre>

<h3>(9) This is the product of the third extension:</h3>
<pre>
<align class="long">
|--5'- /5Acryd//iThioMC6-D/TTTTTTTTAATACGACTCACTATAGGG<tso>AAGCAGTGGTATCAACGCAGAGT</tso>AC<cbc>[10-bp barcode1]</cbc>CAGCTACTGC<cbc>[10-bp barcode2]</cbc>CGAGTACCCT<cbc>[10-bp barcode3]</cbc><umi>[8-bp UMI]</umi>(T)<sub>30</sub> -3'
                       3'- AAAAAAAATTATGCTGAGTGATATCCC<tso>TTCGTCACCATAGTTGCGTCTCA</tso>TG<cbc>[10-bp barcode1]</cbc>GTCGATGACG<cbc>[10-bp barcode2]</cbc>GCTCATGGGA<cbc>[10-bp barcode3]</cbc><umi>[8-bp UMI]</umi>(A)<sub>30</sub> -5'
</align>
</pre>

<h3>(10) Pool again, denature by NaOH, get rid of the bottom strand, neutralise, wash and ready to use (96 x 96 x 96 different combination of barcodes 1, 2 & 3):</h3>
<pre>
<align class="long">
|--5'- /5Acryd//iThioMC6-D/TTTTTTTTAATACGACTCACTATAGGG<tso>AAGCAGTGGTATCAACGCAGAGT</tso>AC<cbc>[10-bp barcode1]</cbc>CAGCTACTGC<cbc>[10-bp barcode2]</cbc>CGAGTACCCT<cbc>[10-bp barcode3]</cbc><umi>[8-bp UMI]</umi>(T)<sub>30</sub> -3'
</align>
</pre>

<br></br>

<h2>Step-by-step library generation</h2>

<h3>(1) Lysis, mRNA anneals to poly-T barcodes on the beads:</h3> 
<pre>
<seq>
5'- XXX...XXXAAAAAAAAAAAAAAA...AAAAAAAAAAAAAAA
        <------TTTTTTTTTTTTTTTTTTTTTTTTTTTTTT<umi>[8-bp UMI]</umi><cbc>[10-bp barcode3]</cbc>TCCCATGAGC<cbc>[10-bp barcode2]</cbc>CGTCATCGAC<cbc>[10-bp barcode1]</cbc>CA<tso>TGAGACGCAACTATGGTGACGAA</tso>GGGATATCACTCAGCATAATTTTTTTT -5'--|
</seq>
</pre>

<h3>(2) The terminal transferase activity of MMLV adds extra Cs:</h3>
<pre>
<seq>
    5'- XXX...XXX(pA)
     CCCXXX...XXX(dT)<umi>[8-bp UMI]</umi><cbc>[10-bp barcode3]</cbc>TCCCATGAGC<cbc>[10-bp barcode2]</cbc>CGTCATCGAC<cbc>[10-bp barcode1]</cbc>CA<tso>TGAGACGCAACTATGGTGACGAA</tso>GGGATATCACTCAGCATAATTTTTTTT -5'--|
</seq>
</pre>

<h3>(3) Perform template switching with TSO:</h3>
<pre>
<align class="long">
5'- <tso>AAGCAGTGGTATCAACGCAGAGT</tso>GAATGGGXXX...XXX(pA)
                  <------------CCCXXX...XXX(dT)<umi>[8-bp UMI]</umi><cbc>[10-bp barcode3]</cbc>TCCCATGAGC<cbc>[10-bp barcode2]</cbc>CGTCATCGAC<cbc>[10-bp barcode1]</cbc>CA<tso>TGAGACGCAACTATGGTGACGAA</tso>GGGATATCACTCAGCATAATTTTTTTT -5'--|
</align>
</pre>

<h3>(4) Break emulsion, purify cDNA:</h3>
<pre>
<align class="long">
3'- <tso>TTCGTCACCATAGTTGCGTCTCA</tso>CTTACCCXXX...XXX(dT)<umi>[8-bp UMI]</umi><cbc>[10-bp barcode3]</cbc>TCCCATGAGC<cbc>[10-bp barcode2]</cbc>CGTCATCGAC<cbc>[10-bp barcode1]</cbc>CA<tso>TGAGACGCAACTATGGTGACGAA</tso>GGGATATCACTCAGCATAATTTTTTTT -5'
</align>
</pre>


<h3>(5) Use the TSO-P for single primer cDNA amplification:<a href="http://www.nature.com/nmeth/journal/v7/n7/full/nmeth.1470.html" target="_blank">( i.e. semi-suppressive PCR )</a>:</h3>
<pre>
<align class="long">
5'- <tso>AAGCAGTGGTATCAACGCAGAGT</tso>--------->
3'- <tso>TTCGTCACCATAGTTGCGTCTCA</tso>CTTACCCXXX...XXX(dT)<umi>[8-bp UMI]</umi><cbc>[10-bp barcode3]</cbc>TCCCATGAGC<cbc>[10-bp barcode2]</cbc>CGTCATCGAC<cbc>[10-bp barcode1]</cbc>CA<tso>TGAGACGCAACTATGGTGACGAA</tso>GGGATATCACTCAGCATAATTTTTTTT -5'
                                                                                                                     <---------<tso>TGAGACGCAACTATGGTGACGAA</tso> -5'
</align>
</pre>

<h3>(5) Purify amplified cDNA:</h3>
<pre>
<align class="long">
5'- <tso>AAGCAGTGGTATCAACGCAGAGT</tso>GAATGGGXXX...XXX(pA)<umi>[8-bp UMI]</umi><cbc>[10-bp barcode3]</cbc>AGGGTACTCG<cbc>[10-bp barcode2]</cbc>GCAGTAGCTG<cbc>[10-bp barcode1]</cbc>GT<tso>ACTCTGCGTTGATACCACTGCTT</tso> -3'
3'- <tso>TTCGTCACCATAGTTGCGTCTCA</tso>CTTACCCXXX...XXX(dT)<umi>[8-bp UMI]</umi><cbc>[10-bp barcode3]</cbc>TCCCATGAGC<cbc>[10-bp barcode2]</cbc>CGTCATCGAC<cbc>[10-bp barcode1]</cbc>CA<tso>TGAGACGCAACTATGGTGACGAA</tso> -5'
</align>
</pre>

<h3>(6) Fragmentation and dA-tailing (three different products):</h3>
<pre>
<seq>
<i>Product 1 (the 5' end of the gene):</i>

5'-  <tso>AAGCAGTGGTATCAACGCAGAGT</tso>GAATGGGXXX...XXXA -3'
3'- A<tso>TTCGTCACCATAGTTGCGTCTCA</tso>CTTACCCXXX...XXX  -5'



<i>Product 2 (the middle part of the gene):</i>

5'-  XXXXXXXXXX...XXXXXXXXXXA -3'
3'- AXXXXXXXXXX...XXXXXXXXXX  -5'



<i>Product 3 (the 3' end of the gene):</i>

5'-  XXX...XXX(pA)<umi>[8-bp UMI]</umi><cbc>[10-bp barcode3]</cbc>AGGGTACTCG<cbc>[10-bp barcode2]</cbc>GCAGTAGCTG<cbc>[10-bp barcode1]</cbc>GT<tso>ACTCTGCGTTGATACCACTGCTT</tso>A -3'
3'- AXXX...XXX(dT)<umi>[8-bp UMI]</umi><cbc>[10-bp barcode3]</cbc>TCCCATGAGC<cbc>[10-bp barcode2]</cbc>CGTCATCGAC<cbc>[10-bp barcode1]</cbc>CA<tso>TGAGACGCAACTATGGTGACGAA</tso>  -5'
</seq>
</pre>

<h3>(7) NEB Hairpin adapter ligation:</h3>
<pre>
<seq>
<i>Product 1 (the 5' end of the gene):</i>

<s5>ACACTCTTTCCCTACACGAC                                                                 ACACGTCTGAACTCCAGTC</s5>\
|                   <s5>GCTCTTCCGATCT</s5><tso>AAGCAGTGGTATCAACGCAGAGT</tso>GAATGGGXXX...XXXA<s5>GATCGGAAGAGC</s5>                   U
U                   <s5>CGAGAAGGCTAG</s5>A<tso>TTCGTCACCATAGTTGCGTCTCA</tso>CTTACCCXXX...XXX<s5>TCTAGCCTTCTCG</s5>                   |
\<s5>CTGACCTCAAGTCTGCACA                                                                 CAGCACATCCCTTTCTCACA</s5>



<i>Product 2 (the middle part of the gene):</i>

<s5>ACACTCTTTCCCTACACGAC                                   ACACGTCTGAACTCCAGTC</s5>\
|                   <s5>GCTCTTCCGATCT</s5>XXX...XXXA<s5>GATCGGAAGAGC</s5>                   U
U                   <s5>CGAGAAGGCTAG</s5>AXXX...XXX<s5>TCTAGCCTTCTCG</s5>                   |
\<s5>CTGACCTCAAGTCTGCACA                                   CAGCACATCCCTTTCTCACA</s5>



<i>Product 3 (the 3' end of the gene):</i>

<s5>ACACTCTTTCCCTACACGAC                                                                                                                                              ACACGTCTGAACTCCAGTC</s5>\
|                   <s5>GCTCTTCCGATCT</s5>XXX...XXX(pA)<umi>[8-bp UMI]</umi><cbc>[10-bp barcode3]</cbc>AGGGTACTCG<cbc>[10-bp barcode2]</cbc>GCAGTAGCTG<cbc>[10-bp barcode1]</cbc>GT<tso>ACTCTGCGTTGATACCACTGCTT</tso>A<s5>GATCGGAAGAGC</s5>                   U
U                   <s5>CGAGAAGGCTAG</s5>AXXX...XXX(dT)<umi>[8-bp UMI]</umi><cbc>[10-bp barcode3]</cbc>TCCCATGAGC<cbc>[10-bp barcode2]</cbc>CGTCATCGAC<cbc>[10-bp barcode1]</cbc>CA<tso>TGAGACGCAACTATGGTGACGAA</tso><s5>TCTAGCCTTCTCG</s5>                   |
\<s5>CTGACCTCAAGTCTGCACA                                                                                                                                              CAGCACATCCCTTTCTCACA</s5>
</seq>
</pre>

<h3>(8) NEB USER Enzyme digestion:</h3>
<pre>
<seq>
<i>Product 1 (the 5' end of the gene, not amplifiable due to the HYi7 PCR primer used in the next step ends with AC):</i>

5'- <s5>ACACTCTTTCCCTACACGAC                                                                 ACACGTCTGAACTCCAGTC</s5> -3'
                        <s5>GCTCTTCCGATCT</s5><tso>AAGCAGTGGTATCAACGCAGAGT</tso>GAATGGGXXX...XXXA<s5>GATCGGAAGAGC</s5>
                        <s5>CGAGAAGGCTAG</s5>A<tso>TTCGTCACCATAGTTGCGTCTCA</tso>CTTACCCXXX...XXX<s5>TCTAGCCTTCTCG</s5>
 3'- <s5>CTGACCTCAAGTCTGCACA                                                                 CAGCACATCCCTTTCTCACA</s5> -5'



<i>Product 2 (the middle part of the gene, not amplifiable due to the primers used in the next step):</i>

5'- <s5>ACACTCTTTCCCTACACGAC                                   ACACGTCTGAACTCCAGTC</s5> -3'
                        <s5>GCTCTTCCGATCT</s5>XXX...XXXA<s5>GATCGGAAGAGC</s5>
                        <s5>CGAGAAGGCTAG</s5>AXXX...XXX<s5>TCTAGCCTTCTCG</s5>
 3'- <s5>CTGACCTCAAGTCTGCACA                                   CAGCACATCCCTTTCTCACA</s5> -5'



<i>Product 3 (the 3' end of the gene, the only amplifiable fragment):</i>

5'- <s5>ACACTCTTTCCCTACACGAC                                                                                                                                              ACACGTCTGAACTCCAGTC</s5> -3'
                        <s5>GCTCTTCCGATCT</s5>XXX...XXX(pA)<umi>[8-bp UMI]</umi><cbc>[10-bp barcode3]</cbc>AGGGTACTCG<cbc>[10-bp barcode2]</cbc>GCAGTAGCTG<cbc>[10-bp barcode1]</cbc>GT<tso>ACTCTGCGTTGATACCACTGCTT</tso>A<s5>GATCGGAAGAGC</s5>
                        <s5>CGAGAAGGCTAG</s5>AXXX...XXX(dT)<umi>[8-bp UMI]</umi><cbc>[10-bp barcode3]</cbc>TCCCATGAGC<cbc>[10-bp barcode2]</cbc>CGTCATCGAC<cbc>[10-bp barcode1]</cbc>CA<tso>TGAGACGCAACTATGGTGACGAA</tso><s5>TCTAGCCTTCTCG</s5>
 3'- <s5>CTGACCTCAAGTCTGCACA                                                                                                                                              CAGCACATCCCTTTCTCACA</s5> -5'
</seq>
</pre>

<h3>(9) Add HYi5_TruSeq and HYi7 primers to perform PCR to amplify the library:</h3>
<pre>
<seq>
<b>Note that only the top strand of Product 3 can be amplified by both primers exponentially:</b>
</seq>
<align class="long">
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>[10-bp index]<s5>ACACTCTTTCCCTACACGACGCT</s5>--------->
                                          5'- <s5>ACACTCTTTCCCTACACGACGCTCTTCCGATCT</s5>XXX...XXX(pA)<umi>[8-bp UMI]</umi><cbc>[10-bp barcode3]</cbc>AGGGTACTCG<cbc>[10-bp barcode2]</cbc>GCAGTAGCTG<cbc>[10-bp barcode1]</cbc>GT<tso>ACTCTGCGTTGATACCACTGCTT</tso>A<s5>GATCGGAAGAGCACACGTCTGAACTCCAGTC</s5> -3'
                                                                                                                                                     <--------------------CA<tso>TGAGACGCAACTATGGTGACGAA</tso>
                                                                                                                                                                                                   GGCGCCTGTC[10-bp index]<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(6) Final library structure:</h3>
<pre>
<align class="small">
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>NNNNNNNNNN<s5>ACACTCTTTCCCTACACGACGCTCTTCCGATCT</s5>XXX...XXX(pA)<umi>NNNNNNNN</umi><cbc>NNNNNNNNNN</cbc>AGGGTACTCG<cbc>NNNNNNNNNN</cbc>GCAGTAGCTG<cbc>NNNNNNNNNN</cbc>GT<tso>ACTCTGCGTTGATACCACTGCTT</tso>CCGCGGACAGNNNNNNNNNN<p7>ATCTCGTATGCCGTCTTCTGCTTG</p7> -3'
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5>NNNNNNNNNN<s5>TGTGAGAAAGGGATGTGCTGCGAGAAGGCTAGA</s5>XXX...XXX(dT)<umi>NNNNNNNN</umi><cbc>NNNNNNNNNN</cbc>TCCCATGAGC<cbc>NNNNNNNNNN</cbc>CGTCATCGAC<cbc>NNNNNNNNNN</cbc>CA<tso>TGAGACGCAACTATGGTGACGAA</tso>GGCGCCTGTCNNNNNNNNNN<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
            <p5>Illumina P5</p5>            10-bp            <s5>TruSeq Read 1</s5>              cDNA        <umi>8-bp</umi>     <cbc>10-bp</cbc>               <cbc>10-bp</cbc>               <cbc>10-bp</cbc>              <tso>TSO</tso>                       10-bp         <p7>Illumina P7</p7>
                               sample index                                                <umi>UMI</umi>     <cbc>barcode3</cbc>            <cbc>barcode2</cbc>            <cbc>barcode1</cbc>                                  sample index
</align>
</pre>

<br></br>

<h2>Library sequencing using Illumina primers</h2>

<h3>(1) Add TruSeq Read 1 sequencing primer to sequence the first read (bottom strand as template, cDNA reads, >50 cycles):</h3>
<pre>
<align class="small">
                                       5'- <s5>ACACTCTTTCCCTACACGACGCTCTTCCGATCT</s5>-------->
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5>NNNNNNNNNN<s5>TGTGAGAAAGGGATGTGCTGCGAGAAGGCTAGA</s5>XXX...XXX(dT)<umi>NNNNNNNN</umi><cbc>NNNNNNNNNN</cbc>TCCCATGAGC<cbc>NNNNNNNNNN</cbc>CGTCATCGAC<cbc>NNNNNNNNNN</cbc>CA<tso>TGAGACGCAACTATGGTGACGAA</tso>GGCGCCTGTCNNNNNNNNNN<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(2) Add HyDrop_CustSeq_Short primer to sequence the sample index at the i7 side (bottom strand as template, 10 cycles):</h3>
<pre>
<align class="small">
                                                                                                                                               5'- GT<tso>ACTCTGCGTTGATACCACTGCTT</tso>CCGCGGACAG--------->
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5>NNNNNNNNNN<s5>TGTGAGAAAGGGATGTGCTGCGAGAAGGCTAGA</s5>XXX...XXX(dT)<umi>NNNNNNNN</umi><cbc>NNNNNNNNNN</cbc>TCCCATGAGC<cbc>NNNNNNNNNN</cbc>CGTCATCGAC<cbc>NNNNNNNNNN</cbc>CA<tso>TGAGACGCAACTATGGTGACGAA</tso>GGCGCCTGTCNNNNNNNNNN<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(3) Add Index 2 sequencing primer to sequence the sample index at the i5 side (top strand as template, 10 cycles):</h3>
<pre>
<align class="small">
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>NNNNNNNNNN<s5>ACACTCTTTCCCTACACGACGCTCTTCCGATCT</s5>XXX...XXX(pA)<umi>NNNNNNNN</umi><cbc>NNNNNNNNNN</cbc>AGGGTACTCG<cbc>NNNNNNNNNN</cbc>GCAGTAGCTG<cbc>NNNNNNNNNN</cbc>GT<tso>ACTCTGCGTTGATACCACTGCTT</tso>CCGCGGACAGNNNNNNNNNN<p7>ATCTCGTATGCCGTCTTCTGCTTG</p7> -3'
                                 <---------<s5>TGTGAGAAAGGGATGTGCTGCGAGAAGGCTAGA</s5> -5'
</align>
</pre>

<h3>(4) Add HyDrop_CustSeq_R2 primer to sequence the second read (top strand as template, 58 cycles, these are cell barcodes and UMI):</h3>
<pre>
<align class="small">
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>NNNNNNNNNN<s5>ACACTCTTTCCCTACACGACGCTCTTCCGATCT</s5>XXX...XXX(pA)<umi>NNNNNNNN</umi><cbc>NNNNNNNNNN</cbc>AGGGTACTCG<cbc>NNNNNNNNNN</cbc>GCAGTAGCTG<cbc>NNNNNNNNNN</cbc>GT<tso>ACTCTGCGTTGATACCACTGCTT</tso>CCGCGGACAGNNNNNNNNNN<p7>ATCTCGTATGCCGTCTTCTGCTTG</p7> -3'
                                                                                         <---------------------------------------------------------CA<tso>TGAGACGCAACTATGGTGACGAA</tso>GGCGCCTGTC -5'
</align>
</pre>

</body>
</html>
