<!DOCTYPE html>
<html lang="en">

<head>
<link rel="stylesheet" type="text/css" href="../style_related/page_format.css">
<title>PETRI-seq</title>
</head>
<body>

<h1><a href="https://www.nature.com/articles/s41564-020-0729-6" target="_blank">PETRI-seq</a></h1>

<p><info><b>PETRI-seq</b> is based on <a href="https://teichlab.github.io/scg_lib_structs/methods_html/SPLiT-seq.html" target="_blank">SPLiT-seq</a>, but it is optimised for prokaryotic single-cell RNA-seq, such as for bacteria. The method uses random priming during the reverse transcription to capture prokaryotic RNA. The procedures are very similar to those in <b>SPLiT-seq</b>.</info></p>
  
<p><info>The <b>PETRI-seq</b> was published in <i>Nature Microbiology</i> <b>5</b>, 1192-1201 (2020). There is also an <a href="https://tavazoielab.c2b2.columbia.edu/PETRI-seq/" target="_blank">accompanying website</a>. The workflow described in this page is based on the protocol from the website, and the oligo sequences are also taken from the website. For a backup, I copied the <a href="../data/PETRI-seq/PETRI-seq_Protocol.pdf" target="_blank">protocol</a> (accessed on 26-Jan-2024) and the <a href="../data/PETRI-seq/PETRI-seq_Oligonucleotide_Table.xlsx" target="_blank">oligo table</a> (accessed on 26-Jan-2024) into this GitHub.</info></p>
  
<br>

<h2>Adapter and primer sequences:</h2>
<seq>
<p><sup>*</sup>Round1 RT Primers (randN): 5'-/5Phos/ <r1>GCCAGA</r1><cbc>[7-bp Round1 barcode]</cbc>NNNNNN -3'</p>
<p><sup>*</sup>Round2 Ligation Oligos: 5'-/5Phos/ <r3>GCTTCGC</r3><cbc>[7-bp Round2 barcode]</cbc><r2>CCTCCTAC</r2> -3'</p>
<p><sup>*</sup>Round3 Ligation Oligos: 5'- AGAA<cs1>TACACGACGCTCTTCCGATCT</cs1><umi>[7-bp UMI]</umi><cbc>[7-bp Round3 barcode]</cbc><r3>GGTCCTTG</r3> -3'</p>
<p>Round 2 barcode linker (SB83): 5'- <r1>STCTGGC</r1><r2>GTAGGAGGW</r2> -3'</p>
<p>Round 3 barcode linker (SB80): 5'- <r3>GCGAAGCCAAGGACCW</r3> -3'</p>
<p>Round 3 blocking strand 1 (SB81): 5'- GCTTCGCTGCAATCGGACCTCGATTGCA -3'</p>
<p>Round 3 blocking strand 2 (SB82): 5'- <r3>WGGTCCTTGGCTTCGC</r3> -3'</p>
<p>Round 2 blocking strand 1 (SB84): 5'- GCCAGASACGTTAGGCAGGACCTAACGT -3'</p>
<p>Round 2 blocking strand 2 (SB85): 5'- <r2>WCCTCCTAC</r2><r1>GCCAGAS</r1> -3'</p>
<p>Nextera Tn5 binding site (19-bp Mosaic End (ME)): 5'- <me>AGATGTGTATAAGAGACAG</me> -3'</p>
<p>Nextera N/S5xx primer entry point (s5): 5'- <s5>TCGTCGGCAGCGTC</s5> -3'</p>
<p>Nextera N7xx primer entry point (s7): 5'- <s7>GTCTCGTGGGCTCGG</s7> -3'</p>
<p>NEB i50x primer (NEB kit E7600): 5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>[8-bp i5 index]<cs1>ACACTCTTTCCCTACACGACGCTCTTCCGATCT</cs1> -3'</p>
<p>Nextera (XT) N7xx Index primer: 5'- <p7>CAAGCAGAAGACGGCATACGAGAT</p7>[8-bp i7 index]<s7>GTCTCGTGGGCTCGG</s7> -3'</p>
<p>Illumina TruSeq Read 1 primer: 5'- <cs1>ACACTCTTTCCCTACACGACGCTCTTCCGATCT</cs1> -3'</p>
<p>Index 1 sequencing primer (i7): 5'- <me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7> -3'</p>
<p>Index 2 sequencing primer (i5): 5'- <cs1>AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT</cs1> -3'</p>
<p>Illumina Nextera Read 2 primer: 5'- <s7>GTCTCGTGGGCTCGG</s7><me>AGATGTGTATAAGAGACAG</me> -3'</p>
<p>Illumina P5 adapter: 5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5> -3'</p>
<p>Illumina P7 adapter: 5'- <p7>CAAGCAGAAGACGGCATACGAGAT</p7> -3'</p>
<p><span style="color:red;"><b>*NOTE:</b></span> The first nt of the <cbc>7-bp Round1 barcode</cbc> is C or G, that is S; The last nt of <cbc>7-bp Round2&3 barcode</cbc> is A or T, that is W.</p>
</seq>


<br>

<h2>Step-by-step library generation</h2>
<h3>(1) Prepare ligation adapters by annealing barcodes with corresponding linkers (in two different plates):</h3>
<pre>
<seq>
<i>Anneal Round2 Ligation Oligos with Round 2 barcode linker (SB83):</i>

5'-/5Phos/ <r3>GCTTCGC</r3><cbc>[7-bp Round2 barcode]</cbc><r2>CCTCCTAC</r2>
                                  3'- <r2>WGGAGGATG</r2><r1>CGGTCTS</r1> -5'


<i>Anneal Round3 Ligation Oligos with Round 3 barcode linker (SB80):</i>

5'- AGAA<cs1>TACACGACGCTCTTCCGATCT</cs1><umi>[7-bp UMI]</umi><cbc>[7-bp Round3 barcode]</cbc><r3>GGTCCTTG</r3>
                                                       3'- <r3>WCCAGGAACCGAAGCG</r3> -5'


</seq>
</pre>

<h3>(2) Anneal Round1 RT primer to mRNA and reverse transcription using Maxima H Minus Reverse Transcriptase <i>in situ</i>:</h3>
<pre>
<seq>
5'-/5Phos/ <r1>GCCAGA</r1><cbc>[7-bp Round1 barcode]</cbc>NNNNNN--->
                           3'- XXXXXXX...XXXXXXX -5'
</seq>
</pre>

<h3>(3) Pool cells and re-distribute to annealed Round2 plate for Round2 Barcode Ligation:</h3>
<pre>
<seq>
5'-/5Phos/ <r3>GCTTCGC</r3><cbc>[7-bp Round2 barcode]</cbc><r2>CCTCCTAC</r2><r1>GCCAGA</r1><cbc>[7-bp Round1 barcode]</cbc>NNNNNN--->
                                  3'- <r2>WGGAGGATG</r2><r1>CGGTCTS</r1> -5'     3'- XXXXXXX...XXXXXXX -5'
</seq>
</pre>

<h3>(4) Pool and re-distribute to annealed Round 3 plate for for Round3 Barcode Ligation:</h3>
<pre>
<seq>
5'- AGAA<cs1>TACACGACGCTCTTCCGATCT</cs1><umi>[7-bp UMI]</umi><cbc>[7-bp Round3 barcode]</cbc><r3>GGTCCTTGGCTTCGC</r3><cbc>[7-bp Round2 barcode]</cbc><r2>CCTCCTAC</r2><r1>GCCAGA</r1><cbc>[7-bp Round1 barcode]</cbc>NNNNNN--->
                                                       3'- <r3>WCCAGGAACCGAAGCG</r3> -5'            3'- <r2>WGGAGGATG</r2><r1>CGGTCTS</r1> -5'     3'- XXXXXXX...XXXXXXX -5'
</seq>
</pre>

<h3>(5) Pool, lyse cells, cDNA purification and 2nd strand synthesis using the NEBNext Second Strand Synthesis system (presumably an RNaseH + DNA pol I based method), and purify the double stranded cDNA:</h3>
<pre>
<seq>
5'- AGAA<cs1>TACACGACGCTCTTCCGATCT</cs1><umi>[7-bp UMI]</umi><cbc>[7-bp Round3 barcode]</cbc><r3>GGTCCTTGGCTTCGC</r3><cbc>[7-bp Round2 barcode]</cbc><r2>CCTCCTAC</r2><r1>GCCAGA</r1><cbc>[7-bp Round1 barcode]</cbc>XXX.XXX -3'
3'- TCTT<cs1>ATGTGCTGCGAGAAGGCTAGA</cs1><umi>[7-bp UMI]</umi><cbc>[7-bp Round3 barcode]</cbc><r3>CCAGGAACCGAAGCG</r3><cbc>[7-bp Round2 barcode]</cbc><r2>GGAGGATG</r2><r1>CGGTCT</r1><cbc>[7-bp Round1 barcode]</cbc>XXX.XXX -5'
</seq>
</pre>

<h3>(6) Tagmentation with Illumina Nextera XT Kit:</h3>
<img src="../data/tn5_dimer.svg" alt="Tn5 dimer" style="width:800px;height:450px;">
<pre>
<align class="long">
<i>Product 1 (s5 at both ends, not amplifiable due to PCR primers used, see the next step):</i>

5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me>
                  <me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s5>CTGCGACGGCTGCT</s5> -5'



<i>Product 2 (s7 at both ends, not amplifiable due to PCR primers used, see the next step):</i>

5'- <s7>GTCTCGTGGGCTCGG</s7><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me>
                   <me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7> -5'



<i>Product 3 (different s5 and s7 at both ends, not amplifiable, due to PCR primers used, see the next step):</i>

5'- <s5>TCGTCGGCAGCGTC</s5><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me>
                  <me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7> -5'


<i>Product 4 (s5 at one end, 3' of cDNA at the other end, not amplifiable due to PCR primers used, see the next step):</i>

5'- AGAA<cs1>TACACGACGCTCTTCCGATCT</cs1><umi>[7-bp UMI]</umi><cbc>[7-bp Round3 barcode]</cbc><r3>GGTCCTTGGCTTCGC</r3><cbc>[7-bp Round2 barcode]</cbc><r2>CCTCCTAC</r2><r1>GCCAGA</r1><cbc>[7-bp Round1 barcode]</cbc>XXX.XXX         <me>CTGTCTCTTATACACATCT</me> -3'
3'- TCTT<cs1>ATGTGCTGCGAGAAGGCTAGA</cs1><umi>[7-bp UMI]</umi><cbc>[7-bp Round3 barcode]</cbc><r3>CCAGGAACCGAAGCG</r3><cbc>[7-bp Round2 barcode]</cbc><r2>GGAGGATG</r2><r1>CGGTCT</r1><cbc>[7-bp Round1 barcode]</cbc>XXX.XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s5>CTGCGACGGCTGCT</s5> -5'


<i>Product 5 (s7 at one end, 3' of cDNA at the other end, the only amplifiable fragment, see the next step):</i>

5'- AGAA<cs1>TACACGACGCTCTTCCGATCT</cs1><umi>[7-bp UMI]</umi><cbc>[7-bp Round3 barcode]</cbc><r3>GGTCCTTGGCTTCGC</r3><cbc>[7-bp Round2 barcode]</cbc><r2>CCTCCTAC</r2><r1>GCCAGA</r1><cbc>[7-bp Round1 barcode]</cbc>XXX.XXX         <me>CTGTCTCTTATACACATCT</me> -3'
3'- TCTT<cs1>ATGTGCTGCGAGAAGGCTAGA</cs1><umi>[7-bp UMI]</umi><cbc>[7-bp Round3 barcode]</cbc><r3>CCAGGAACCGAAGCG</r3><cbc>[7-bp Round2 barcode]</cbc><r2>GGAGGATG</r2><r1>CGGTCT</r1><cbc>[7-bp Round1 barcode]</cbc>XXX.XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7> -5'


</align>
</pre>

<h3>(7) Use NEB i50X and Nextera XT N7xx primers for library amplification:</h3>
<pre>
<align class="long">
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>[i5]<cs1>ACACTCTTTCCCTACACGACGCTCTTCCGATCT</cs1>-------->
                                         5'- AGAA<cs1>TACACGACGCTCTTCCGATCT</cs1><umi>[7-bp UMI]</umi><cbc>[7-bp Round3 barcode]</cbc><r3>GGTCCTTGGCTTCGC</r3><cbc>[7-bp Round2 barcode]</cbc><r2>CCTCCTAC</r2><r1>GCCAGA</r1><cbc>[7-bp Round1 barcode]</cbc>XXX.XXX         <me>CTGTCTCTTATACACATCT</me> -3'
                                         3'- TCTT<cs1>ATGTGCTGCGAGAAGGCTAGA</cs1><umi>[7-bp UMI]</umi><cbc>[7-bp Round3 barcode]</cbc><r3>CCAGGAACCGAAGCG</r3><cbc>[7-bp Round2 barcode]</cbc><r2>GGAGGATG</r2><r1>CGGTCT</r1><cbc>[7-bp Round1 barcode]</cbc>XXX.XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7> -5'
                                                                                                                                                                                                          <----<s7>GGCTCGGGTGCTCTG</s7>[i7]<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(8) Final library structure:</h3>
<pre>
<align class="long">
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>NNNNNNNN<cs1>ACACTCTTTCCCTACACGACGCTCTTCCGATCT</cs1><umi>NNNNNNN</umi><cbc>NNNNNNN</cbc><r3>GGTCCTTGGCTTCGC</r3><cbc>NNNNNNN</cbc><r2>CCTCCTAC</r2><r1>GCCAGA</r1><cbc>NNNNNNN</cbc>XXX.XXX<me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7>NNNNNNNN<p7>ATCTCGTATGCCGTCTTCTGCTTG</p7> -3'
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5>NNNNNNNN<cs1>TGTGAGAAAGGGATGTGCTGCGAGAAGGCTAGA</cs1><umi>NNNNNNN</umi><cbc>NNNNNNN</cbc><r3>CCAGGAACCGAAGCG</r3><cbc>NNNNNNN</cbc><r2>GGAGGATG</r2><r1>CGGTCT</r1><cbc>NNNNNNN</cbc>XXX.XXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7>NNNNNNNN<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
            <p5>Illumina P5</p5>           8bp i5          <cs1>TruSeq Read 1</cs1>            <umi>7-bp</umi>   <cbc>7-bp</cbc>      <r3>Round3</r3>      <cbc>7-bp</cbc>      <r2>Round2</r2>     <cbc>7-bp</cbc>   cDNA          <me>ME</me>              <s7>s7</s7>         8bp i7       <p7>Illumina P7</p7>
                                                                           <umi>UMI</umi>   <cbc>Round3</cbc>     <r3>linker</r3>     <cbc>Round2</cbc>     <r2>linker</r2>    <cbc>Round1</cbc>
                                                                                 <cbc>barcode</cbc>               <cbc>barcode</cbc>              <cbc>barcode</cbc>
</align>
</pre>

<br>

<h2>Library sequencing:</h2>

<h3>(1) Add Illumina TruSeq Read 1 sequencing primer to sequence the first read (bottom strand as template, UMI and barcode reads, at least 57 cycles):</h3>
<pre>
<align class="long">
                                     5'- <cs1>ACACTCTTTCCCTACACGACGCTCTTCCGATCT</cs1>-------------------------------------------------------->
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5>NNNNNNNN<cs1>TGTGAGAAAGGGATGTGCTGCGAGAAGGCTAGA</cs1><umi>NNNNNNN</umi><cbc>NNNNNNN</cbc><r3>CCAGGAACCGAAGCG</r3><cbc>NNNNNNN</cbc><r2>GGAGGATG</r2><r1>CGGTCT</r1><cbc>NNNNNNN</cbc>XXX.XXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7>NNNNNNNN<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(2) Add Index 1 sequencing primer to sequence i7 index (bottom strand as template, 8 cycles):</h3>
<pre>
<align class="long">
                                                                                                                                      5'- <me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7>------->
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5>NNNNNNNN<cs1>TGTGAGAAAGGGATGTGCTGCGAGAAGGCTAGA</cs1><umi>NNNNNNN</umi><cbc>NNNNNNN</cbc><r3>CCAGGAACCGAAGCG</r3><cbc>NNNNNNN</cbc><r2>GGAGGATG</r2><r1>CGGTCT</r1><cbc>NNNNNNN</cbc>XXX.XXX<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7>NNNNNNNN<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(3) Cluster regeneration, add Index 2 sequencing primer to sequence the i5 index (top strand as template, 8 cycles):</h3>
<pre>
<align class="long">
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>NNNNNNNN<cs1>ACACTCTTTCCCTACACGACGCTCTTCCGATCT</cs1><umi>NNNNNNN</umi><cbc>NNNNNNN</cbc><r3>GGTCCTTGGCTTCGC</r3><cbc>NNNNNNN</cbc><r2>CCTCCTAC</r2><r1>GCCAGA</r1><cbc>NNNNNNN</cbc>XXX.XXX<me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7>NNNNNNNN<p7>ATCTCGTATGCCGTCTTCTGCTTG</p7> -3'
                                 <-------<cs1>TGTGAGAAAGGGATGTGCTGCGAGAAGGCTAGA</cs1>
</align>
</pre>

<h3>(4) Add Illumina Read 2 sequencing primer to sequence the second read (top strand as template, cDNA read, 17 cycles):</h3>
<pre>
<align class="long">
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>NNNNNNNN<cs1>ACACTCTTTCCCTACACGACGCTCTTCCGATCT</cs1><umi>NNNNNNN</umi><cbc>NNNNNNN</cbc><r3>GGTCCTTGGCTTCGC</r3><cbc>NNNNNNN</cbc><r2>CCTCCTAC</r2><r1>GCCAGA</r1><cbc>NNNNNNN</cbc>XXX.XXX<me>CTGTCTCTTATACACATCT</me><s7>CCGAGCCCACGAGAC</s7>NNNNNNNN<p7>ATCTCGTATGCCGTCTTCTGCTTG</p7> -3'
                                                                                                                                   <------<me>GACAGAGAATATGTGTAGA</me><s7>GGCTCGGGTGCTCTG</s7> -5'
</align>
</pre>

</body>
</html>
