<!DOCTYPE html>
<html lang="en">

<head>
<link rel="stylesheet" type="text/css" href="../style_related/page_format.css">
<title>scifi-ATAC-seq</title>
</head>
<body>

<h1><a href="https://www.biorxiv.org/content/10.1101/2023.09.17.558155v3" target="_blank">scifi-ATAC-seq</a></h1>

<p><span style="font-size:1.1em;">The idea of scifi-ATAC-seq is very similar to <a href="https://teichlab.github.io/scg_lib_structs/methods_html/dscATAC.html" target="_blank">dsciATAC-seq</a> and <a href="https://teichlab.github.io/scg_lib_structs/methods_html/txci-ATAC-seq.html" target="_blank">txci-ATAC-seq</a>. It works on the <a href="https://teichlab.github.io/scg_lib_structs/methods_html/10xChromium_scATAC.html" target="_blank">10X Genomics Single Cell ATAC</a> platform, but it does require custom sequencing primer and recipe. The procedure was based on the information from the preprint.</span></p>

<br>

<h2>Adapter and primer sequences:</h2>

<seq>
<p>10x Genomics Beads-oligo: |--5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5><cbc>[16-bp GEM barcode]</cbc><s5>TCGTCGGCAGCGTC</s5> -3'</p>
<p>Tn5-ME-A: 5'- <s5>TCGTCGGCAGCGTC</s5>GATATGTGATAATGAGGAC<cbc>[4-bp Tn5 index A]</cbc><me>AGATGTGTATAAGAGACAG</me> -3'</p>
<p>Tn5-ME-B: 5'- <s7>GTCTCGTGGGCTCGG</s7>TGAATGTGTAGAAGACAGA<cbc>[4-bp Tn5 index B]</cbc><me>AGATGTGTATAAGAGACAG</me> -3'</p>
<p>Tn5ME-bottom: 5'-/phos/ <me>CTGTCTCTTATACACATCT</me> -3'</p>
<p>Nextera Tn5 binding site (19-bp Mosaic End (ME)): 5'- <me>AGATGTGTATAAGAGACAG</me> -3'</p>
<p>Nextera S5xx primer entry point (s5): 5'- <s5>TCGTCGGCAGCGTC</s5> -3'</p>
<p>Nextera N7xx primer entry point (s7): 5'- <s7>GTCTCGTGGGCTCGG</s7> -3'</p>
<p>i7 Sample Index Plate N, Set A (PN-3000262): 5'- <p7>CAAGCAGAAGACGGCATACGAGAT</p7>[8-bp sample index]<s7>GTCTCGTGGGCTCGG</s7> -3'</p>
<p>Custom Read 1 sequencing primer (1_Read1): 5'- <s5>TCGTCGGCAGCGTC</s5>GATATGTGATAATGAGGAC -3'</p>
<p>Custom i7 index sequencing primer (2_Index1): 5'- TCTGTCTTCTACACATTCA<s7>CCGAGCCCACGAGAC</s7> -3'</p>
<p>Custom i5 index sequencing primer (3_Index2): 5'- GTCCTCATTATCACATATC<s5>GACGCTGCCGACGA</s5> -3'</p>
<p>Custom Read 2 sequencing primer (4_Read2): 5'- <s7>GTCTCGTGGGCTCGG</s7>TGAATGTGTAGAAGACAGA -3'</p>
<p>Illumina P5 adapter: 5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5> -3'</p>
<p>Illumina P7 adapter: 5'- <p7>CAAGCAGAAGACGGCATACGAGAT</p7> -3'</p>
</seq>

<br>

<h2>Step-by-step library generation:</h2>
<h3>(1) Anneal "Tn5ME-bottom+Tn5-ME-A" and "Tn5ME-bottom+Tn5-ME-B" and use those to assemble the indexed Tn5 transposomes:</h3>
<img src="../data/scifi-ATAC-seq_Tn5.svg" alt="Tn5 dimer" style="width:750px;">

<h3>(2) Bulk nuclei tagging by the indexed Tn5 shown above. There are 3 different products (will create 9 bp gap):</h3>
<pre>
<seq>
<i>Product 1 (s5 at both ends, not amplifiable due to <a href="http://www.nature.com/nmeth/journal/v7/n7/full/nmeth.1470.html" target="_blank">semi-suppressiev PCR</a>):</i>

5'- <s5>TCGTCGGCAGCGTC</s5>GATATGTGATAATGAGGAC<cbc>[4-bp Tn5 index A]</cbc><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me>
                                                       <me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><cbc>[4-bp Tn5 index A]</cbc>CAGGAGTAATAGTGTATAG<s5>CTGCGACGGCTGCT</s5> -5'


<i>Product 2 (s7 at both ends, not amplifiable due to <a href="http://www.nature.com/nmeth/journal/v7/n7/full/nmeth.1470.html" target="_blank">semi-suppressiev PCR</a>):</i>

5'- <s7>GTCTCGTGGGCTCGG</s7>TGAATGTGTAGAAGACAGA<cbc>[4-bp Tn5 index B]</cbc><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me>
                                                        <me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><cbc>[4-bp Tn5 index B]</cbc>AGACAGAAGATGTGTAAGT<s7>GGCTCGGGTGCTCTG</s7> -5'


<i>Product 3 (different ends, amplifiable):</i>

5'- <s5>TCGTCGGCAGCGTC</s5>GATATGTGATAATGAGGAC<cbc>[4-bp Tn5 index A]</cbc><me>AGATGTGTATAAGAGACAG</me>XXXXXXXXXXXX...XXX         <me>CTGTCTCTTATACACATCT</me>
                                                       <me>TCTACACATATTCTCTGTC</me>         XXX...XXXXXXXXXXXX<me>GACAGAGAATATGTGTAGA</me><cbc>[4-bp Tn5 index B]</cbc>AGACAGAAGATGTGTAAGT<s7>GGCTCGGGTGCTCTG</s7> -5'
</seq>
</pre>

<h3>(3) Load to 10X machine for GEM barcode addition:</h3>
<pre>
<seq>
<i>Gap fill-in (72 C, 5 mins):</i>

5'- <s5>TCGTCGGCAGCGTC</s5>GATATGTGATAATGAGGAC<cbc>[4-bp Tn5 index A]</cbc><me>AGATGTGTATAAGAGACAG</me>XXX...XXX<me>CTGTCTCTTATACACATCT</me><cbc>[4-bp Tn5 index B]</cbc>TCTGTCTTCTACACATTCA<s7>CCGAGCCCACGAGAC</s7> -3'
3'- <s5>AGCAGCCGTCGCAG</s5>CTATACACTATTACTCCTG<cbc>[4-bp Tn5 index A]</cbc><me>TCTACACATATTCTCTGTC</me>XXX...XXX<me>GACAGAGAATATGTGTAGA</me><cbc>[4-bp Tn5 index B]</cbc>AGACAGAAGATGTGTAAGT<s7>GGCTCGGGTGCTCTG</s7> -5'


<i>Single-primer linear PCR for GEM barcode addition:</i>

|--5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5><cbc>[16-bp GEM barcode]</cbc><s5>TCGTCGGCAGCGTC</s5>---------->
                                                   5'- <s5>TCGTCGGCAGCGTC</s5>GATATGTGATAATGAGGAC<cbc>[4-bp Tn5 index A]</cbc><me>AGATGTGTATAAGAGACAG</me>XXX...XXX<me>CTGTCTCTTATACACATCT</me><cbc>[4-bp Tn5 index B]</cbc>TCTGTCTTCTACACATTCA<s7>CCGAGCCCACGAGAC</s7> -3'
                                                   3'- <s5>AGCAGCCGTCGCAG</s5>CTATACACTATTACTCCTG<cbc>[4-bp Tn5 index A]</cbc><me>TCTACACATATTCTCTGTC</me>XXX...XXX<me>GACAGAGAATATGTGTAGA</me><cbc>[4-bp Tn5 index B]</cbc>AGACAGAAGATGTGTAAGT<s7>GGCTCGGGTGCTCTG</s7> -5'
</seq>
</pre>

<h3>(4) Break emulsion and DNA purification. This is the product from above:</h3>
<pre>
<seq>
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5><cbc>[16-bp GEM barcode]</cbc><s5>TCGTCGGCAGCGTC</s5>GATATGTGATAATGAGGAC<cbc>[4-bp Tn5 index A]</cbc><me>AGATGTGTATAAGAGACAG</me>XXX...XXX<me>CTGTCTCTTATACACATCT</me><cbc>[4-bp Tn5 index B]</cbc>TCTGTCTTCTACACATTCA<s7>CCGAGCCCACGAGAC</s7>
    <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5><cbc>[16-bp GEM barcode]</cbc><s5>AGCAGCCGTCGCAG</s5>CTATACACTATTACTCCTG<cbc>[4-bp Tn5 index A]</cbc><me>TCTACACATATTCTCTGTC</me>XXX...XXX<me>GACAGAGAATATGTGTAGA</me><cbc>[4-bp Tn5 index B]</cbc>AGACAGAAGATGTGTAAGT<s7>GGCTCGGGTGCTCTG</s7> -5'
</seq>
</pre>

<h3>(5) Sample Index PCR using Illumina P5 and i7 Sample Index Plate N (PN-3000262):</h3>
<pre>
<align class="long">
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5>------------------->
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5><cbc>[16-bp GEM barcode]</cbc><s5>TCGTCGGCAGCGTC</s5>GATATGTGATAATGAGGAC<cbc>[4-bp Tn5 index A]</cbc><me>AGATGTGTATAAGAGACAG</me>XXX...XXX<me>CTGTCTCTTATACACATCT</me><cbc>[4-bp Tn5 index B]</cbc>TCTGTCTTCTACACATTCA<s7>CCGAGCCCACGAGAC</s7>
    <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5><cbc>[16-bp GEM barcode]</cbc><s5>AGCAGCCGTCGCAG</s5>CTATACACTATTACTCCTG<cbc>[4-bp Tn5 index A]</cbc><me>TCTACACATATTCTCTGTC</me>XXX...XXX<me>GACAGAGAATATGTGTAGA</me><cbc>[4-bp Tn5 index B]</cbc>AGACAGAAGATGTGTAAGT<s7>GGCTCGGGTGCTCTG</s7> -5'
                                                                                                                                                                          <----------------<s7>GGCTCGGGTGCTCTG</s7>[8-bp sample index]<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(6) Final library structure:</h3>
<pre>
<align class="long">
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5><cbc>NNNNNNNNNNNNNNNN</cbc><s5>TCGTCGGCAGCGTC</s5>GATATGTGATAATGAGGAC<cbc>NNNN</cbc><me>AGATGTGTATAAGAGACAG</me>XXX...XXX<me>CTGTCTCTTATACACATCT</me><cbc>NNNN</cbc>TCTGTCTTCTACACATTCA<s7>CCGAGCCCACGAGAC</s7>NNNNNNNN<p7>ATCTCGTATGCCGTCTTCTGCTTG</p7>
    <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5><cbc>NNNNNNNNNNNNNNNN</cbc><s5>AGCAGCCGTCGCAG</s5>CTATACACTATTACTCCTG<cbc>NNNN</cbc><me>TCTACACATATTCTCTGTC</me>XXX...XXX<me>GACAGAGAATATGTGTAGA</me><cbc>NNNN</cbc>AGACAGAAGATGTGTAAGT<s7>GGCTCGGGTGCTCTG</s7>NNNNNNNN<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
             <p5>Illumina P5</p5>              <cbc>16 bp</cbc>           <s5>s5</s5>                          <cbc>4 bp</cbc>       <me>ME</me>            gDNA          <me>ME</me>          <cbc>4 bp</cbc>                         <s7>s7</s7>         8 bp        <p7>Illumina P7</p7>
                                   <cbc>GEM barcode</cbc>                                 <cbc>Tn5 Index A</cbc>                                        <cbc>Tn5 Index B</cbc>                            sample index
</align>
</pre>

<br>

<h2>Library sequencing:</h2>

<h3>(1) Add 1_Read1 sequencing primer to sequence the first read (presumably 74 cycles, bottom strand as template, the first 4 bp are the Tn5 Index A, the last 51 bp are gDNA, the middle 19 bp are ME):</h3>
<pre>
<align class="long">
                                             5'- <s5>TCGTCGGCAGCGTC</s5>GATATGTGATAATGAGGAC------------------------->
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5><cbc>NNNNNNNNNNNNNNNN</cbc><s5>AGCAGCCGTCGCAG</s5>CTATACACTATTACTCCTG<cbc>NNNN</cbc><me>TCTACACATATTCTCTGTC</me>XXX...XXX<me>GACAGAGAATATGTGTAGA</me><cbc>NNNN</cbc>AGACAGAAGATGTGTAAGT<s7>GGCTCGGGTGCTCTG</s7>NNNNNNNN<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(2) Add 2_Index1 sequencing primer to sequence sample index (i7) (bottom strand as template, 8 cycles):</h3>
<pre>
<align class="long">
                                                                                                                                     5'- TCTGTCTTCTACACATTCA<s7>CCGAGCCCACGAGAC</s7>------->
3'- <p5>TTACTATGCCGCTGGTGGCTCTAGATGTG</p5><cbc>NNNNNNNNNNNNNNNN</cbc><s5>AGCAGCCGTCGCAG</s5>CTATACACTATTACTCCTG<cbc>NNNN</cbc><me>TCTACACATATTCTCTGTC</me>XXX...XXX<me>GACAGAGAATATGTGTAGA</me><cbc>NNNN</cbc>AGACAGAAGATGTGTAAGT<s7>GGCTCGGGTGCTCTG</s7>NNNNNNNN<p7>TAGAGCATACGGCAGAAGACGAAC</p7> -5'
</align>
</pre>

<h3>(3) Cluster regeneration, add 3_Index2 sequencing primer to sequence the second index (i5) (top strand as template, 16 cycles, this is the GEM barcode):</h3>
<pre>
<align class="long">
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5><cbc>NNNNNNNNNNNNNNNN</cbc><s5>TCGTCGGCAGCGTC</s5>GATATGTGATAATGAGGAC<cbc>NNNN</cbc><me>AGATGTGTATAAGAGACAG</me>XXX...XXX<me>CTGTCTCTTATACACATCT</me><cbc>NNNN</cbc>TCTGTCTTCTACACATTCA<s7>CCGAGCCCACGAGAC</s7>NNNNNNNN<p7>ATCTCGTATGCCGTCTTCTGCTTG</p7>
                                 <---------------<s5>AGCAGCCGTCGCAG</s5>CTATACACTATTACTCCTG -5'
</align>
</pre>

<h3>(4) Add 4_Read2 sequencing primer to sequence the second read (top strand as template, presumably 74 cycles, the first 4 bp are the Tn5 Index B, the last 51 bp are gDNA, the middle 19 bp are ME):</h3>
<pre>
<align class="long">
5'- <p5>AATGATACGGCGACCACCGAGATCTACAC</p5><cbc>NNNNNNNNNNNNNNNN</cbc><s5>TCGTCGGCAGCGTC</s5>GATATGTGATAATGAGGAC<cbc>NNNN</cbc><me>AGATGTGTATAAGAGACAG</me>XXX...XXX<me>CTGTCTCTTATACACATCT</me><cbc>NNNN</cbc>TCTGTCTTCTACACATTCA<s7>CCGAGCCCACGAGAC</s7>NNNNNNNN<p7>ATCTCGTATGCCGTCTTCTGCTTG</p7>
                                                                                                             <---------------------------AGACAGAAGATGTGTAAGT<s7>GGCTCGGGTGCTCTG</s7> -5'
</align>
</pre>

<br>

</body>
</html>
